---
title: "Data Availability"
output:
  html_document:
    code_folding: none
    toc: false
---

<p id = "steel">
To run the workflow, install the appropriate R packages and download this file:
</p>

[10.6084/m9.figshare.7357178](https://doi.org/10.6084/m9.figshare.7357178){target="_new"}: DOI for this workflow. Includes output from the DADA2 workflow, the phyloseq script, and other necessary input files.

<div style="padding-top: 1em"></div>

<p id = "steel">
We also offer several additional data products and processing scripts.
</p>


* [10.6084/m9.figshare.6875522](https://doi.org/10.6084/m9.figshare.6875522){target="_new"}: Raw data for each sample (before removing primers) plus two tables for ENA submission.
* [PRJEB28397](https://www.ebi.ac.uk/ena/browser/view/PRJEB28397){target="_new"}: Study accession number for trimmed data (primers removed)  deposited at the European Nucleotide Archive.
* [10.6084/m9.figshare.6997253](https://doi.org/10.6084/m9.figshare.6997253){target="_new"}: DADA2 workflow for processing 16S rRNA reads.
* [10.6084/m9.figshare.7379930](https://doi.org/10.6084/m9.figshare.7379930){target="_new"}: Data products from the workflows including sequence tables, taxonomy tables, and ASV fasta files.
* [10.6084/m9.figshare.7379936](https://doi.org/10.6084/m9.figshare.7379936){target="_new"}: Fasta files for the 59 DA ASVs and top BLAST results, plus the alignment file (including top BLAST hits).
* [10.6084/m9.figshare.7379597](https://doi.org/10.6084/m9.figshare.7379597){target="_new"}: Supplementary file from the paper.

<div style="padding-top: 1em"></div>

<p id = "steel">
Accessing the R Code only.
</p>

The  R code is available by clicking on the code button in the menu bar or [here](raw_code.txt). Please note that this  R code is pulled from all the `.Rmd` files. This has not been tested independent of the R Markdown workflows so <span class="paper">Use at Your Own Risk</span>. In other words, the code works in the Rmarkdown format but the complete pipeline has not been tested using just this code. I used `knitr::purl()` to pull  the code from the Rmarkdown file. I did this for you just in case you wanted the code and not hear me drone on about colors, zoomable figures, or Keanu Reeves. The first part is the DADA2 workflow and the second part is the phyloseq workflow. Commands that are commented out are things I tried that I could never get to work. Any line that starts like this: `## ----` is the code chuck name and details.
