<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Part 3: Data Preparation</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.8/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.16/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.16/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DIGEST</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="0_pw_intro.html">
    <span class="fa fa-info"></span>
     
    Intro
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-wrench"></span>
     
    Workflow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_field_observations.html">1. Field Observations</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Processing</li>
    <li>
      <a href="2_dada2.html">2. DADA2 Workflow</a>
    </li>
    <li>
      <a href="3_data_prep.html">3. Data Prep</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Analyses</li>
    <li>
      <a href="4_diversity.html">4. Diversity</a>
    </li>
    <li>
      <a href="5_da_asv.html">5. DA ASVs</a>
    </li>
    <li>
      <a href="6_synthesis.html">6. Synthesis</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="7_appendices.html">Appendices</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="data_availability.html">
    <span class="fa fa-database"></span>
     
    Data
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-user-friends"></span>
     
    People
  </a>
</li>
<li>
  <a href="supplemental_material.html">SOM</a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-phone"></span>
     
    Contact
  </a>
</li>
<li>
  <a href="how_to_build_this_site.html">
    <span class="fa fa-question fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Part 3: Data Preparation</h1>

</div>


<p><a href="0_pw_intro.html#workflow_overview">Back to Introduction</a></p>
<hr />
<div id="define-groups" class="section level2">
<h2>Define groups</h2>
<div class="notice">
<p>Now then, on to the microbial data. First, we load the data packet produced by the final step of the DADA2 workflow (Combine_Runs.Rmd), format sample names, and define groupings. We will use the sample names to define the different groups.</p>
</div>
<p>
</p>
<pre class="r"><code>load(&quot;DATA/PHYLOSEQ/combo_pipeline.rdata&quot;)
samples.out &lt;- rownames(seqtab)
subject &lt;- sapply(strsplit(samples.out, &quot;[[:digit:]]&quot;), `[`, 1)
# this splits the string at first instance of a digit
# use the whole string for individuals
# use the first two letters for genus
# use the next three letters for species
sample_name &lt;- substr(samples.out, 1, 999)
genus &lt;- substr(samples.out, 1, 2)
species &lt;- substr(samples.out, 1, 5)</code></pre>
<blockquote>
<p>Groups</p>
</blockquote>
<ul>
<li>53 individuals</li>
<li>3 genera</li>
<li>7 species</li>
</ul>
<p>And finally we define a sample data frame that holds the different groups we extracted from the sample names. On the right are a few samples and their different groups names.</p>
<pre class="r"><code>#define a sample data frame
samdf &lt;- data.frame(SamName = sample_name, Gen = genus, Sp = species)
rownames(samdf) &lt;- samples.out
kable(samdf[c(1, 13, 20, 30, 44), 1:3], row.names = FALSE) %&gt;%
  kable_styling(bootstrap_options = &quot;striped&quot;,
                full_width = FALSE, position = &quot;float_right&quot;)  %&gt;%
  column_spec(1:3, width = &quot;3.5cm&quot;)</code></pre>
<table class="table table-striped" style="width: auto !important; float: right; margin-left: 10px;">
<thead>
<tr>
<th style="text-align:left;">
SamName
</th>
<th style="text-align:left;">
Gen
</th>
<th style="text-align:left;">
Sp
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 3.5cm; ">
AcCoe01
</td>
<td style="text-align:left;width: 3.5cm; ">
Ac
</td>
<td style="text-align:left;width: 3.5cm; ">
AcCoe
</td>
</tr>
<tr>
<td style="text-align:left;width: 3.5cm; ">
AcTra05
</td>
<td style="text-align:left;width: 3.5cm; ">
Ac
</td>
<td style="text-align:left;width: 3.5cm; ">
AcTra
</td>
</tr>
<tr>
<td style="text-align:left;width: 3.5cm; ">
ScTae03
</td>
<td style="text-align:left;width: 3.5cm; ">
Sc
</td>
<td style="text-align:left;width: 3.5cm; ">
ScTae
</td>
</tr>
<tr>
<td style="text-align:left;width: 3.5cm; ">
SpAur02
</td>
<td style="text-align:left;width: 3.5cm; ">
Sp
</td>
<td style="text-align:left;width: 3.5cm; ">
SpAur
</td>
</tr>
<tr>
<td style="text-align:left;width: 3.5cm; ">
SpVir07
</td>
<td style="text-align:left;width: 3.5cm; ">
Sp
</td>
<td style="text-align:left;width: 3.5cm; ">
SpVir
</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Abbreviations:</p>
</blockquote>
<ul>
<li>AcCoe = <em>Acanthurus coeuleus</em></li>
<li>AcTra = <em>Acanthurus tractus</em></li>
<li>ScTae = <em>Scarus taeniopterus</em></li>
<li>SpAur = <em>Sparisoma aurofrenatum</em></li>
<li>SpVir = <em>Sparisoma viride</em></li>
<li>ScVet = <em>Scarus vetula</em></li>
<li>SpChr = <em>Sparisoma chrysopterum</em></li>
</ul>
</div>
<div id="phyloseq-objects" class="section level2">
<h2>Phyloseq objects</h2>
<p>Next we create a phyloseq (ps) object with the Silva (slv) taxonomy. There is also a Greengenes (gg) annotation in the output file from DADA2 which can be used instead of the Silva annotation. Just change <code>tax_silva</code> to <code>tax_gg</code>. At this point we rename the amplicon sequence variants (ASVs) so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention—ASV1, ASV2, ASV3, and so on, while retaining the exact sequences.</p>
<p>The phyloseq object looks like this:</p>
<pre class="r"><code># this create the phyloseq object
ps_slv &lt;- phyloseq(otu_table(seqtab, taxa_are_rows = FALSE),
                   sample_data(samdf), tax_table(tax_silva))
tax_table(ps_slv) &lt;- cbind(tax_table(ps_slv), rownames(tax_table(ps_slv)))
# adding unique ASV names
taxa_names(ps_slv) &lt;- paste0(&quot;ASV&quot;, seq(ntaxa(ps_slv)))
tax_table(ps_slv) &lt;- cbind(tax_table(ps_slv), rownames(tax_table(ps_slv)))
ps_slv</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 12479 taxa and 53 samples ]
## sample_data() Sample Data:       [ 53 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 12479 taxa by 8 taxonomic ranks ]</code></pre>
<p>While the ASV names look like this: ASV1, ASV2, ASV3, ASV4, ASV5, ASV6 and so on…</p>
<p>At this point we have a completely unadulterated phyloseq object because it contains all ASVs and all samples. We add two final columns with the actual ASV sequences and ASV IDs. This will be useful later when trying to export a fasta file. Finally, we export the sequence and taxonomy tables, for posterity sake.</p>
<pre class="r"><code>colnames(tax_table(ps_slv)) &lt;- c(&quot;Kingdom&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;,
                                 &quot;Family&quot;, &quot;Genus&quot;, &quot;ASV_SEQ&quot;, &quot;ASV_ID&quot;)
write.table(tax_table(ps_slv), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/full_tax_table.txt&quot;,
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(t(otu_table(ps_slv)), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/full_seq_table.txt&quot;,
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(sample_data(ps_slv), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/full_sample_data.txt&quot;,
            sep = &quot;\t&quot;, quote = FALSE, row.names =  FALSE)</code></pre>
<p>Remember three of these samples were omitted because we did not have replicates for the host species. Lets remove those samples. The only way we could figure out how to do this was by selecting the samples we <em>wanted to keep</em>. If you want to change the group of samples, modify the script accordingly.</p>
<pre class="r"><code>ps_slv_base &lt;- prune_samples(c(&quot;SpAur01&quot;, &quot;SpAur02&quot;, &quot;SpAur03&quot;, &quot;SpAur04&quot;,
                               &quot;SpAur10&quot;, &quot;SpAur11&quot;, &quot;SpAur12&quot;, &quot;SpAur13&quot;,
                               &quot;SpVir01&quot;, &quot;SpVir02&quot;, &quot;SpVir03&quot;, &quot;SpVir04&quot;,
                               &quot;SpVir05&quot;, &quot;SpVir06&quot;, &quot;SpVir07&quot;, &quot;SpVir08&quot;,
                               &quot;SpVir09&quot;, &quot;SpVir10&quot;, &quot;SpVir11&quot;, &quot;AcCoe01&quot;,
                               &quot;AcCoe02&quot;, &quot;AcCoe03&quot;, &quot;AcCoe04&quot;, &quot;AcCoe05&quot;,
                               &quot;AcCoe06&quot;, &quot;AcCoe07&quot;, &quot;AcCoe08&quot;, &quot;AcTra01&quot;,
                               &quot;AcTra02&quot;, &quot;AcTra03&quot;, &quot;AcTra04&quot;, &quot;AcTra05&quot;,
                               &quot;AcTra06&quot;, &quot;AcTra07&quot;, &quot;AcTra08&quot;, &quot;AcTra09&quot;,
                               &quot;ScTae01&quot;, &quot;ScTae02&quot;, &quot;ScTae03&quot;, &quot;ScTae04&quot;,
                               &quot;ScTae05&quot;, &quot;ScTae06&quot;, &quot;ScTae07&quot;, &quot;ScTae08&quot;,
                               &quot;ScTae09&quot;, &quot;SpAur05&quot;, &quot;SpAur06&quot;, &quot;SpAur07&quot;,
                               &quot;SpAur08&quot;, &quot;SpAur09&quot;), ps_slv)
ps_slv_base</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 12479 taxa and 50 samples ]
## sample_data() Sample Data:       [ 50 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 12479 taxa by 8 taxonomic ranks ]</code></pre>
<p>0K, three samples gone. But we probably lost some ASVs when use we removed samples. So we need to get rid of any ASVs that have now a total of <strong>0 reads</strong>. This will be our working phyloseq object.</p>
<pre class="r"><code>ps_slv_work &lt;- prune_taxa(taxa_sums(ps_slv_base) &gt; 0, ps_slv_base)
ps_slv_work</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 12040 taxa and 50 samples ]
## sample_data() Sample Data:       [ 50 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 12040 taxa by 8 taxonomic ranks ]</code></pre>
<p>Great, there were <strong>439 ASVs</strong> found only in those three samples.</p>
<p>We can also export seq and tax tables for our trimmed dataset and get a quick summary of the trimmed dataset before removing unwanted reads. .</p>
<pre class="r"><code>write.table(tax_table(ps_slv_work), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/trim_tax_table.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(t(otu_table(ps_slv_work)), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/trim_seq_table.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(sample_data(ps_slv_work), &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/trim_sample_data.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, row.names =  FALSE)

# general stats for the dataset.

sample_sum_df_raw &lt;- data.frame(sum = sample_sums(ps_slv_work))
total_reads_raw &lt;- sum(otu_table(ps_slv_work))
smin_raw &lt;- as.integer(min(sample_sums(ps_slv_work)))
smean_raw &lt;- as.integer(mean(sample_sums(ps_slv_work)))
smax_raw &lt;- as.integer(max(sample_sums(ps_slv_work)))</code></pre>
<p>Looks like the total number of reads in the dataset (after removing unwanted samples) is <strong>3120211</strong>; range of 12178 to 182696 reads per sample and an average of 62404 reads per sample.</p>
</div>
<div id="remove-contaminants" class="section level2">
<h2>Remove contaminants</h2>
<p>These samples are intestinal communities and we assume that Chloroplast are not contributing to metabolism. These data could be useful later but for now lets create a phyloseq object without Chloroplast.</p>
<p><font size="3" color="red">WARNING</font>: the <code>subset_taxa</code> command removes anything that is <code>NA</code> for the specified taxonomic level or above. <strong>For example</strong>, lets say you run the <code>subset_taxa</code> command using <code>Order != &quot;Chloroplast&quot;</code>. Seems like you should get a phyloseq object with everything except Chloroplast. But actually the command not only gets rid Chloroplast but everything else that has <code>NA</code> for Order and above. In our experience this is not well documented and we had to dig through the files to figure out what was happening.</p>
<p>Our dataset has 590 Chloroplast ASVs and running the command as is removed an additional 1244 ASVs. So lets see if we can get rid of just Chloroplast ASVs without removing everything that is unclassified at Order and above. To do this, we subset the taxa to generate a ps object of just Chloroplast, selected the ASV column only, turned it into a factor, and used this to remove Chloroplast from the ps object.</p>
<pre class="r"><code># generate a file with Chloroplast ASVs
chloro_p_ps &lt;- subset_taxa(ps_slv_work, Order == &quot;Chloroplast&quot;)
chloro_p_tab &lt;-  as(tax_table(chloro_p_ps), &quot;matrix&quot;)
chloro_p_tab &lt;- chloro_p_tab[, 8]
chloro_p_df &lt;- as.factor(chloro_p_tab)
goodTaxaCH &lt;- setdiff(taxa_names(ps_slv_work), chloro_p_df)
ps_slv_work_no_cyano &lt;- prune_taxa(goodTaxaCH, ps_slv_work)
ps_slv_work_no_cyano</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 11450 taxa and 50 samples ]
## sample_data() Sample Data:       [ 50 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 11450 taxa by 8 taxonomic ranks ]</code></pre>
<pre class="r"><code>###### Summarize data
total_asv_chloro_p &lt;- length(chloro_p_df)

sample_sum_chloro_p_ps &lt;- data.frame(sum = sample_sums(chloro_p_ps))
total_reads_chloro_p_ps &lt;- sum(otu_table(chloro_p_ps))
smin_chloro_p_ps &lt;- min(sample_sums(chloro_p_ps))
smean_chloro_p_ps &lt;- mean(sample_sums(chloro_p_ps))
smax_chloro_p_ps &lt;- max(sample_sums(chloro_p_ps))

write.table(tax_table(chloro_p_ps), 
            &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/chloroplast_tax_table.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(t(otu_table(chloro_p_ps)), 
            &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/chloroplast_seq_table.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, col.names = NA)
write.table(sample_data(chloro_p_ps), 
            &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/PS/chloroplast_sample_data.txt&quot;, 
            sep = &quot;\t&quot;, quote = FALSE, row.names =  FALSE)</code></pre>
<p>This step removed <strong>590 Chloroplast ASVs</strong> encompassing <strong>149274 total reads</strong>. Perfect.</p>
<p>And now we use the same approach to remove Mitochondria.</p>
<pre class="r"><code># generate a file with mitochondria ASVs
MT1_ps &lt;- subset_taxa(ps_slv_work_no_cyano, Family == &quot;Mitochondria&quot;)
MT1 &lt;-  as(tax_table(MT1_ps), &quot;matrix&quot;)
MT1 &lt;- MT1[, 8]
MT1df &lt;- as.factor(MT1)
goodTaxa &lt;- setdiff(taxa_names(ps_slv_work_no_cyano), MT1df)
ps_slv_work_filt &lt;- prune_taxa(goodTaxa, ps_slv_work_no_cyano)
ps_slv_work_filt</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 11144 taxa and 50 samples ]
## sample_data() Sample Data:       [ 50 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 11144 taxa by 8 taxonomic ranks ]</code></pre>
<pre class="r"><code>###### Summarize data
total_asv_MT1 &lt;- length(MT1df)
# colnames(tax_table(MT1_ps))
sample_sum_MT1_ps &lt;- data.frame(sum = sample_sums(MT1_ps))
total_reads_MT1_ps &lt;- sum(otu_table(MT1_ps))
smin_MT1_ps &lt;- min(sample_sums(MT1_ps))
smean_MT1_ps &lt;- mean(sample_sums(MT1_ps))
smax_MT1_ps &lt;- max(sample_sums(MT1_ps))</code></pre>
<p>Sweet, looks like this removed <strong>306 Mitochondria ASVs</strong> encompassing <strong>35941 total reads</strong>.</p>
<pre class="r"><code># general stats for the dataset.
#colnames(tax_table(ps_slv_work_filt))
sample_sum_df &lt;- data.frame(sum = sample_sums(ps_slv_work_filt))
total_reads &lt;- sum(otu_table(ps_slv_work_filt))
smin &lt;- as.integer(min(sample_sums(ps_slv_work_filt)))
smean &lt;- as.integer(mean(sample_sums(ps_slv_work_filt)))
smax &lt;- as.integer(max(sample_sums(ps_slv_work_filt)))</code></pre>
<p>After removing contaminants here is what the final dataset looks like:</p>
<ul>
<li>Total number of reads in the dataset is <strong>2934996</strong>.</li>
<li>Range of <strong>11686</strong> to <strong>180159</strong> reads per sample.</li>
<li>Average of <strong>58699</strong> reads per sample.</li>
</ul>
</div>
<div id="merged-phyloseq-object" class="section level2">
<h2>Merged phyloseq object</h2>
<p>One last thing to do is to create a merged phyloseq object where samples are grouped by host species. This will come in handy later for some analyses.</p>
<pre class="r"><code>mergedGP &lt;- merge_samples(ps_slv_work_filt, &quot;Sp&quot;)
SD &lt;- merge_samples(sample_data(ps_slv_work_filt), &quot;Sp&quot;)
mergedGP</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 11144 taxa and 5 samples ]
## sample_data() Sample Data:       [ 5 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 11144 taxa by 8 taxonomic ranks ]</code></pre>
<p>Great, still the same number of ASVs and now only 5 “samples” corresponding to the 5 species: AcCoe, AcTra, ScTae, SpAur, SpVir.</p>
<p>There are now the several phyloseq objects to chose from and, using the above methods, additional objects can easily be created.</p>
<ul>
<li><code>ps_slv</code> –&gt; phyloseq dataset with all 53 samples, all ASVs.</li>
<li><code>ps_slv_base</code> –&gt; phyloseq dataset with 50 samples, all ASVs (this is not very useful).</li>
<li><code>ps_slv_work</code> –&gt; phyloseq dataset with 50 samples, zero-read ASVs removed.</li>
<li><code>ps_slv_work_filt</code> –&gt; phyloseq dataset with 50 samples, ASVs and reads from Mitochondria and Chloroplast removed.</li>
<li><code>mergedGP</code> –&gt; <code>ps_slv_work_filt</code> phyloseq dataset collapsed by host species.</li>
</ul>
<p>Here we can save some phyloseq objects so we can make some Shiny Apps at some point.</p>
<pre class="r"><code>save(ps_slv_work_filt, mergedGP, file = &quot;DATA/PHYLOSEQ/PS_OBJECTS/ps.RData&quot;)
saveRDS(ps_slv_work_filt, &quot;DATA/PHYLOSEQ/PS_OBJECTS/ps_slv_work_filt.rds&quot;)
saveRDS(mergedGP, &quot;DATA/PHYLOSEQ/PS_OBJECTS/mergedGP.rds&quot;)</code></pre>
</div>
<div id="host-information" class="section level2">
<h2>Host Information</h2>
<p>Before we do anything else, lets generate summary data for each host. We can generate a summary report for any ps object but we will use the object with mitochondria and chlorplasts removed, as well as the low replicate host species removed. We will also add details about each host. The table is displayed below. We can use these data when we upload the original fastq files to sequence read archives. Later on we will also add alpha diversity stats and save the table.</p>
<hr />
<div id="host-details" class="section level4">
<h4>Host details</h4>
<pre class="r"><code>total_reads &lt;- sample_sums(ps_slv_work_filt)
total_reads &lt;- as.data.frame(total_reads, make.names = TRUE)
total_reads &lt;- total_reads %&gt;% rownames_to_column(&quot;host_ID&quot;)

total_asvs &lt;- estimate_richness(ps_slv_work_filt, measures = &quot;Observed&quot;)
total_asvs &lt;- total_asvs %&gt;% rownames_to_column(&quot;host_ID&quot;)

sam_details &lt;- sample_data(ps_slv)
sam_details &lt;- sam_details %&gt;% mutate(genus = case_when(
    Gen == &quot;Ac&quot; ~ &quot;Acanthurus&quot;, 
    Gen == &quot;Sc&quot; ~ &quot;Scarus&quot;,
    Gen == &quot;Sp&quot; ~ &quot;Sparisoma&quot;))

sam_details &lt;- sam_details %&gt;% mutate(species = case_when(
    Sp == &quot;AcCoe&quot;~ &quot;coeruleus&quot;,
    Sp == &quot;AcTra&quot;~ &quot;tractus&quot;,
    Sp == &quot;ScTae&quot;~ &quot;taeniopterus&quot;,
    Sp == &quot;SpAur&quot;~ &quot;aurofrenatum&quot;,
    Sp == &quot;SpVir&quot;~ &quot;viride&quot;))
#Sp == &quot;SpChr&quot;~ &quot;chrysopterum&quot;,
#Sp == &quot;ScVet&quot;~ &quot;vetula&quot;
sam_details &lt;- sam_details %&gt;% mutate(common_name = case_when(
    Sp == &quot;AcCoe&quot; ~ &quot;blue tang surgeonfish&quot;,
    Sp == &quot;AcTra&quot; ~ &quot;fiveband surgeonfish&quot;,
    Sp == &quot;ScTae&quot; ~ &quot;princess parrotfish&quot;,
    Sp == &quot;SpAur&quot; ~ &quot;redband parrotfish&quot;, 
    Sp == &quot;SpVir&quot; ~ &quot;stoplight parrotfish&quot;))

#Sp == &quot;SpChr&quot; ~ &quot;redtail parrotfish&quot;,
#Sp == &quot;ScVet&quot; ~ &quot;queen parrotfish&quot;))

sam_details &lt;- sam_details %&gt;% mutate(NCBI_txid = case_when(
    Sp == &quot;AcCoe&quot; ~ &quot;157585&quot;,
    Sp == &quot;AcTra&quot; ~ &quot;1316013&quot;,
    Sp == &quot;ScTae&quot; ~ &quot;544418&quot;,
    Sp == &quot;SpAur&quot; ~ &quot;59663&quot;,
    Sp == &quot;SpVir&quot; ~ &quot;59666&quot;))
#Sp == &quot;SpChr&quot; ~ &quot;51766&quot;,
#Sp == &quot;ScVet&quot; ~ &quot;84543&quot;))

sam_details &lt;- sam_details[-c(2, 3)]
colnames(sam_details) &lt;- c(&quot;host_ID&quot;, &quot;host_genus&quot;,
                           &quot;host_species&quot;, &quot;full_name&quot;,
                           &quot;NCBI_txid&quot;)

merge_tab &lt;- merge(sam_details, total_reads, by = &quot;host_ID&quot;)
merge_tab2 &lt;- merge(merge_tab, total_asvs, by = &quot;host_ID&quot;)
colnames(merge_tab2) &lt;- c(&quot;host_ID&quot;, &quot;host_genus&quot;,
                          &quot;host_species&quot;, &quot;common_name&quot;,
                          &quot;NCBI_txid&quot;,  &quot;total_reads&quot;,
                          &quot;total_ASVs&quot;)

# We also have a datatable containing metrics for each host. Lets bring this in 
# and merge with  the summary table
metrics &lt;- read.table(&quot;DATA/PHYLOSEQ/TABLES/INPUT/host_metrics.txt&quot;, 
                      sep = &quot;\t&quot;, header = TRUE)
host_details &lt;- merge(merge_tab2, metrics, by = &quot;host_ID&quot;)
colnames(host_details) &lt;- c(&quot;host_ID&quot;, &quot;host_genus&quot;, &quot;host_species&quot;,
                            &quot;common_name&quot;, &quot;NCBI_txid&quot;,  &quot;total_reads&quot;,
                            &quot;total_ASVs&quot;, &quot;collection_date&quot;, &quot;phase&quot;,
                            &quot;weight&quot;, &quot;total_length&quot;, &quot;foregut_length&quot;,
                            &quot;midgut_length&quot;, &quot;hindgut_length&quot;,
                            &quot;total_gut_length&quot;)

datatable(host_details, rownames = FALSE, width = &quot;100%&quot;,
          colnames = c(&quot;host_ID&quot;, &quot;host_genus&quot;, &quot;host_species&quot;,
                       &quot;common_name&quot;, &quot;NCBI_txid&quot;,  &quot;total_reads&quot;,
                       &quot;total_ASVs&quot;, &quot;Collection_date&quot;, &quot;Phase&quot;, 
                       &quot;Weight (g)&quot;, &quot;Total length (cm)&quot;, 
                       &quot;Fore gut length (cm)&quot;, &quot;Mid gut length (cm)&quot;, 
                       &quot;Hind gut length (cm)&quot;, &quot;Total gut length (cm)&quot;), 
          caption = htmltools::tags$caption(style = &quot;caption-side: 
                                            bottom; text-align: left;&quot;, 
                                            &quot;Table: &quot;, 
                                            htmltools::em(&quot;Sample summary.&quot;)), 
          extensions = &quot;Buttons&quot;, 
          options = list(columnDefs = 
                           list(list(className = &quot;dt-left&quot;, targets = 0)), 
                         dom = &quot;Blfrtip&quot;, pageLength = 5, 
                         lengthMenu = c(5, 10, 25, 50), 
                         buttons = c(&quot;csv&quot;, &quot;copy&quot;), 
                         scrollX = TRUE, scrollCollapse = TRUE))</code></pre>
<div id="htmlwidget-c862fc109d4a4d313929" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c862fc109d4a4d313929">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: &#10;                                            bottom; text-align: left;\">\n  Table: \n  <em>Sample summary.<\/em>\n<\/caption>","data":[["AcCoe01","AcCoe02","AcCoe03","AcCoe04","AcCoe05","AcCoe06","AcCoe07","AcCoe08","AcTra01","AcTra02","AcTra03","AcTra04","AcTra05","AcTra06","AcTra07","AcTra08","AcTra09","ScTae01","ScTae02","ScTae03","ScTae04","ScTae05","ScTae06","ScTae07","ScTae08","ScTae09","SpAur01","SpAur02","SpAur03","SpAur04","SpAur05","SpAur06","SpAur07","SpAur08","SpAur09","SpAur10","SpAur11","SpAur12","SpAur13","SpVir01","SpVir02","SpVir03","SpVir04","SpVir05","SpVir06","SpVir07","SpVir08","SpVir09","SpVir10","SpVir11"],["Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma"],["coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","viride","viride","viride","viride","viride","viride","viride","viride","viride","viride","viride"],["blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish"],["157585","157585","157585","157585","157585","157585","157585","157585","1316013","1316013","1316013","1316013","1316013","1316013","1316013","1316013","1316013","544418","544418","544418","544418","544418","544418","544418","544418","544418","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59666","59666","59666","59666","59666","59666","59666","59666","59666","59666","59666"],[64360,137180,105966,180159,58982,105859,78401,117863,31483,53448,47620,65285,88469,60784,48607,131146,57571,52898,36997,28642,80074,58809,53118,82019,42598,27312,42513,11686,43164,15359,45483,19398,37006,13220,28047,71679,26893,62031,87183,39112,71391,75561,19366,73304,20724,49961,44363,41082,74298,26522],[1148,630,702,461,827,789,475,402,534,194,793,1503,2031,1768,488,391,1516,795,771,676,1308,723,576,1022,297,786,175,154,153,332,187,612,257,96,486,592,198,915,350,470,1037,1548,418,598,666,594,665,634,572,108],["2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-20","2016-7-20","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-20","2016-7-20","2016-7-20","2016-7-20","2016-7-20","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-4-24","2016-4-24","2016-4-24","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19"],["large","large","large","large","large","large","large","juvenile","large","small","large","large","large","large","small","small","large","terminal","initial","terminal","terminal","terminal","terminal","initial","initial","initial","terminal","initial","terminal","initial","initial","initial","initial","terminal","initial","initial","juvenile","terminal","initial","initial","terminal","terminal","initial","initial","initial","initial","initial","initial","initial","initial"],[266,187.8,129.9,188.1,277.5,235.8,226,8,89.8,69,101.5,88.5,98,84,13,23,96,526,115,331,574,394.6,445.8,220.9,142.04,195.2,236,112,212,114,null,null,null,185.1,152.6,175.05,29.4,173.09,142.5,358,null,1124,408.41,564,552.8,310.1,300.45,645,243.5,150.5],[23.5,20,17.5,20.5,23,22.5,22,7,17.2,15,17,16.5,18.4,16,8.3,9.8,17.5,30.5,19,25.4,30.3,27.9,28.7,22,19.6,21.5,24.5,18,23.1,18.5,null,null,null,21.6,19.5,21.8,11.5,21,20,27,40,43,28.5,34,33.8,25.5,26,38,24.2,19.5],[17,9,7,9,15,10,14,4,7,10,10,null,4,null,3.5,5,5,11.5,7,8,19,10.5,10,9,4,8.5,11,9,7,5,null,null,null,8.8,9,10.5,3.1,7.2,6,10,16,null,11,null,12.5,12.8,10.7,13,6,null],[64,74,71,76,97,83,71,22,53,43,32,null,49,null,24,29,53,31.5,20,25,39,31.5,34.1,29,22,27,18,17,26,17,null,null,null,25.8,26.1,28.2,8.7,27.8,20.5,39,78,null,39,null,53,46,42.4,65.5,34,null],[9,12,4,7,7,5,7,4.5,6,3,4,null,3,null,3,4,4,5,1.5,3,4,5,3.5,2.6,3,4,4,2,3.5,5,null,null,null,3,3,3,0.7,2.6,3,3.5,10,null,5,null,4.8,2.5,4.3,4.7,3,null],[90,95,82,92,119,98,92,30.5,66,56,46,null,56,null,30.5,38,62,48,28.5,36,62,47,47.6,40.6,29,39.5,33,28,36.5,27,null,null,null,37.6,38.1,41.7,12.5,37.6,29.5,52.5,104,null,55,null,70.3,61.3,57.4,83.2,43,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>host_ID<\/th>\n      <th>host_genus<\/th>\n      <th>host_species<\/th>\n      <th>common_name<\/th>\n      <th>NCBI_txid<\/th>\n      <th>total_reads<\/th>\n      <th>total_ASVs<\/th>\n      <th>Collection_date<\/th>\n      <th>Phase<\/th>\n      <th>Weight (g)<\/th>\n      <th>Total length (cm)<\/th>\n      <th>Fore gut length (cm)<\/th>\n      <th>Mid gut length (cm)<\/th>\n      <th>Hind gut length (cm)<\/th>\n      <th>Total gut length (cm)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[5,6,9,10,11,12,13,14]}],"dom":"Blfrtip","pageLength":5,"lengthMenu":[5,10,25,50],"buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="warning">
<p>This table also scrolls horizontally.</p>
</div>
<p>Now we have a nice little summary table about each sample—genus/species, common name, number of reads, number of ASVs, etc. All of this info can be used when submitting samples to sequence read archives. Once we conduct alpha diversity estimates below, we will add that data to the table above and export as <strong>Supplementary Table 3</strong>.</p>
</div>
<div id="go-to-part-4-community-composition-diversity" class="section level4">
<h4><a href="4_diversity.html">Go to Part 4, Community Composition &amp; Diversity</a></h4>
</div>
</div>

<footer role="contentinfo" id="site-footer">

<hr>

<!-- /.bottom-menu -->
	<p class="copyright">&#169; 2019 metacrobe,
		powered by <a href="http://www.r-project.org">R</a> + <a href="http://rmarkdown.rstudio.com">RMarkdown</a> + <a href="http://github.com">Github</a>.</p>
</footer>

<!--this prevents a white space at bottom of page -->
<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;"></div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
