<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Part 4: Community Composition &amp; Diversity</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.8/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.16/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.16/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DIGEST</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="0_pw_intro.html">
    <span class="fa fa-info"></span>
     
    Intro
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-wrench"></span>
     
    Workflow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_field_observations.html">1. Field Observations</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Processing</li>
    <li>
      <a href="2_dada2.html">2. DADA2</a>
    </li>
    <li>
      <a href="3_data_prep.html">3. Data Prep</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Analyses</li>
    <li>
      <a href="4_diversity.html">4. Diversity</a>
    </li>
    <li>
      <a href="5_da_asv.html">5. DA ASVs</a>
    </li>
    <li>
      <a href="6_synthesis.html">6. Synthesis</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="7_appendices.html">Appendices</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="data_availability.html">
    <span class="fa fa-database"></span>
     
    Data
  </a>
</li>
<li>
  <a href="supplemental_material.html">SOM</a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-phone"></span>
     
    Contact
  </a>
</li>
<li>
  <a href="how_to_build_this_site.html">
    <span class="fa fa-question fa-lg"></span>
     
    About
  </a>
</li>
<li>
  <a href="https://github.com/projectdigest/web/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="raw_code.txt">
    <span class="fa fa-code fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="people.html">
    <span class="fa fa-user-friends"></span>
     
    People
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Part 4: Community Composition &amp; Diversity</h1>

</div>


<p><a href="0_pw_intro.html#workflow_overview">Back to Introduction</a></p>
<hr />
<div class="notice">
<p>What are the dominant taxa in this system? How diverse are these communities? How similar are samples to each other? Here we look at <strong>a</strong>) Taxonomic diversity, <strong>b</strong>) alpha-diversity, and <strong>c</strong>) beta-diversity.</p>
</div>
<p>
</p>
<hr />
<div id="taxonomic-composition" class="section level2">
<h2>Taxonomic Composition</h2>
<div id="total-reads-asvs" class="section level3">
<h3>Total reads &amp; ASVs</h3>
<p>Before we can start to understand a system, we need to know something about its parts. So lets start with a quick look at Class-level diversity. Of course, you can change this to any taxonomic rank you wish. Here we created a <strong>sortable</strong> table that has the total number of reads and ASVs for each class</p>
<pre class="r"><code># generate the ASV table
tax_asv &lt;- table(tax_table(ps_slv_work_filt)[, &quot;Class&quot;], 
                 exclude = NULL, dnn = &quot;Taxa&quot;)
tax_asv &lt;- as.data.frame(tax_asv, make.names = TRUE)

#### change &lt;NA&gt; to Unclassified
# Get levels and add &quot;None&quot;
levels &lt;- levels(tax_asv$Taxa)
levels[length(levels) + 1] &lt;- &quot;Unclassified&quot;
# refactor Taxa to include &quot;Unclassified&quot; as a factor level
# and replace NA with &quot;Unclassified&quot;
tax_asv$Taxa &lt;- factor(tax_asv$Taxa, levels = levels)
tax_asv$Taxa[is.na(tax_asv$Taxa)] &lt;- &quot;Unclassified&quot;

# generate the reads table
tax_reads &lt;- factor(tax_table(ps_slv_work_filt)[, &quot;Class&quot;], exclude = NULL)
tax_reads &lt;- apply(otu_table(ps_slv_work_filt), MARGIN = 1, 
                   function(x) {
                     tapply(x, INDEX = tax_reads, 
                                       FUN = sum, na.rm = FALSE, 
                                       simplify = TRUE)})

#RENAME NA --&gt; Unclassified
rownames(tax_reads)[72] &lt;- &quot;Unclassified&quot;

tax_reads &lt;- as.data.frame(tax_reads, make.names = TRUE)
tax_reads &lt;- cbind(tax_reads, reads = rowSums(tax_reads))
#DELETE all but last column
tax_reads &lt;- tax_reads[51]
tax_reads &lt;- setDT(tax_reads, keep.rownames = TRUE)[]
# merge the two tables and make everything look pretty
# in an interactive table

taxa_read_asv_tab &lt;- merge(tax_reads, tax_asv, by.x = &quot;rn&quot;, by.y = &quot;Taxa&quot;)
taxa_read_asv_tab &lt;- mutate(taxa_read_asv_tab, 
                            prop_of_ASVs = Freq / sum(Freq), 
                            prop_of_reads = reads / sum(reads))
taxa_read_asv_tab &lt;- taxa_read_asv_tab[c(1, 2, 5, 3, 4)]

names(taxa_read_asv_tab) &lt;- c(&quot;Class&quot;, &quot;total_reads&quot;, &quot;prop_of_reads&quot;, 
                              &quot;total_ASVs&quot;, &quot;prop_of_ASVs&quot;)

taxa_read_asv_tab2 &lt;- taxa_read_asv_tab
taxa_read_asv_tab2$prop_of_reads &lt;- round(taxa_read_asv_tab2$prop_of_reads, 
                                          digits = 6)
taxa_read_asv_tab2$prop_of_ASVs &lt;- round(taxa_read_asv_tab2$prop_of_ASVs, 
                                         digits = 6)</code></pre>
<p>For your information, the dataset has a total of <strong>2934996</strong> reads across <strong>11144</strong> ASVs.</p>
<div class="paper">
Table S4
</div>
<pre class="r"><code>#kills sci notation     
options(scipen = 999) 
write.table(taxa_read_asv_tab2, &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/SUPP/Table_S4.txt&quot;, 
            sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE)

datatable(taxa_read_asv_tab2, rownames = FALSE, width = &quot;100%&quot;, 
          colnames = c(&quot;Class&quot;, &quot;total_reads&quot;, &quot;prop_of_reads&quot;, 
                       &quot;total_ASVs&quot;, &quot;prop_of_ASVs&quot;), 
          caption = 
            htmltools::tags$caption(
              style = &quot;caption-side: bottom; text-align: left;&quot;, 
              &quot;Table: &quot;, 
              htmltools::em(&quot;Total reads &amp; ASVs by Class&quot;)), 
          extensions = &quot;Buttons&quot;, 
          options = list(columnDefs = 
                           list(list(className = &quot;dt-left&quot;, targets = 0)), 
                         dom = &quot;Blfrtip&quot;, pageLength = 5, 
                         lengthMenu = c(5, 10, 35, 70), 
                         buttons = c(&quot;csv&quot;, &quot;copy&quot;)))</code></pre>
<div id="htmlwidget-c862fc109d4a4d313929" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c862fc109d4a4d313929">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: left;\">\n  Table: \n  <em>Total reads &amp; ASVs by Class<\/em>\n<\/caption>","data":[["028H05-P-BN-P5","Acidimicrobiia","Acidobacteriia","Actinobacteria","Alphaproteobacteria","Anaerolineae","Babeliae","Bacilli","Bacteroidia","BD2-11_terrestrial_group","BD7-11","Blastocatellia_(Subgroup_4)","Calditrichia","Campylobacteria","Chlamydiae","Chlorobia","Clostridia","Coriobacteriia","Dadabacteriia","Deferribacteres","Dehalococcoidia","Deinococci","Deltaproteobacteria","Entotheonellia","Erysipelotrichia","Fibrobacteria","Fusobacteriia","Gammaproteobacteria","Gracilibacteria","Halanaerobiia","Holophagae","Hydrogenedentia","Ignavibacteria","KD4-96","Kiritimatiellae","KIST-JJY010","Ktedonobacteria","Lentisphaeria","Lineage_IIb","Melainabacteria","Microgenomatia","Moduliflexia","Mollicutes","NC10","Negativicutes","Nitrososphaeria","Nitrospira","Oligosphaeria","OM190","Oxyphotobacteria","Parcubacteria","PAUC43f_marine_benthic_group","Phycisphaerae","Pla3_lineage","Pla4_lineage","Planctomycetacia","Rhodothermia","Saccharimonadia","Spirochaetia","Subgroup_17","Subgroup_22","Subgroup_26","Subgroup_6","Subgroup_9","Synergistia","Thermoanaerobaculia","Thermoleophilia","Unclassified","vadinHA49","Verrucomicrobiae","WWE3","Zetaproteobacteria"],[25,3261,30,4726,315809,787,579,3331,217335,13,40,29,2,9451,85,236,302028,8114,3459,708,29,249,219331,3,83548,8,260519,1040757,30,42,45,2,6,158,668,11,5,2416,4,775,11,6,33447,3,97,364,35,697,2839,71304,2,8,3223,93,12,262356,160,705,28375,150,441,274,29,571,7,1084,147,7027,662,42087,3,123],[9e-06,0.001111,1e-05,0.00161,0.107601,0.000268,0.000197,0.001135,0.07405,4e-06,1.4e-05,1e-05,1e-06,0.00322,2.9e-05,8e-05,0.102906,0.002765,0.001179,0.000241,1e-05,8.5e-05,0.07473,1e-06,0.028466,3e-06,0.088763,0.354603,1e-05,1.4e-05,1.5e-05,1e-06,2e-06,5.4e-05,0.000228,4e-06,2e-06,0.000823,1e-06,0.000264,4e-06,2e-06,0.011396,1e-06,3.3e-05,0.000124,1.2e-05,0.000237,0.000967,0.024294,1e-06,3e-06,0.001098,3.2e-05,4e-06,0.089389,5.5e-05,0.00024,0.009668,5.1e-05,0.00015,9.3e-05,1e-05,0.000195,2e-06,0.000369,5e-05,0.002394,0.000226,0.01434,1e-06,4.2e-05],[3,81,5,51,1978,52,88,77,1349,3,8,3,1,4,16,1,597,3,29,3,2,6,570,1,54,1,44,1664,11,5,3,1,3,8,24,1,2,13,1,35,2,2,110,1,6,23,5,4,191,449,1,2,245,16,3,2512,13,53,16,15,42,10,2,26,1,64,7,198,24,295,1,4],[0.000269,0.007268,0.000449,0.004576,0.177495,0.004666,0.007897,0.00691,0.121052,0.000269,0.000718,0.000269,9e-05,0.000359,0.001436,9e-05,0.053571,0.000269,0.002602,0.000269,0.000179,0.000538,0.051149,9e-05,0.004846,9e-05,0.003948,0.149318,0.000987,0.000449,0.000269,9e-05,0.000269,0.000718,0.002154,9e-05,0.000179,0.001167,9e-05,0.003141,0.000179,0.000179,0.009871,9e-05,0.000538,0.002064,0.000449,0.000359,0.017139,0.040291,9e-05,0.000179,0.021985,0.001436,0.000269,0.225413,0.001167,0.004756,0.001436,0.001346,0.003769,0.000897,0.000179,0.002333,9e-05,0.005743,0.000628,0.017767,0.002154,0.026472,9e-05,0.000359]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>total_reads<\/th>\n      <th>prop_of_reads<\/th>\n      <th>total_ASVs<\/th>\n      <th>prop_of_ASVs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[1,2,3,4]}],"dom":"Blfrtip","pageLength":5,"lengthMenu":[5,10,35,70],"buttons":["csv","copy"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Looks like Proteobacteria, Firmicutes, Fusobacteria, Planctomycetes, and Bacteroidetes dominate in the read department. Curiously, Fusobacteria has comparatively low ASV richness.</p>
<hr />
<p>Much of the analyses we do from here on out will be at the <strong>Class</strong> &amp; <strong>Family</strong> levels. We chose not to focus on the Genus level because there simply is not enough resolution in our dataset to build a cohesive story. This is because these fish are (microbially) understudied <em>and</em> we are dealing with short read data. On the other hand, Phylum level is too coarse for groups like Proteobacteria and Firmicutes. Order did not provide any additional information and can be cumbersome for taxa with poorly resolved lineages. Depending on the dataset, you may want to change your strategy.</p>
<p>Lets take a closer look Class-level taxonomic content of these communities. There are numerous ways to do this but here we chose to collapse samples by host species and display the relative abundance of the most dominant taxa. We also generated alternative views of taxonomic composition for individual samples—a box-and-whisker plot as well as bar plots separated by host and taxa. The figures and code are available in <a href="1_field_observations.html">Appendix A</a>.</p>
<p>Stacked bar charts are not the best but we like them for a birds eye view of the data. Here we calculate the relative abundance of taxa for each host species at the <strong>Class</strong> level. It turns out this is not too easy in phyloseq and there is a lot of (messy) code.</p>
<pre class="r"><code># calculate the averages and merge by species
ps_slv_filt_AVG &lt;- transform_sample_counts(ps_slv_work_filt, 
                                           function(x) x / sum(x))
mergedGP_BAR &lt;- merge_samples(ps_slv_filt_AVG, &quot;Sp&quot;)
SD_BAR &lt;- merge_samples(sample_data(ps_slv_filt_AVG), &quot;Sp&quot;)

# merge taxa by rank. If you choose a different rank be sure to change
# the rank throughout this code chunk
mdata_phy &lt;- tax_glom(mergedGP_BAR, taxrank = &quot;Class&quot;, NArm = FALSE)  
mdata_phyrel &lt;- transform_sample_counts(mdata_phy, function(x) x / sum(x))
meltd &lt;- psmelt(mdata_phyrel)
meltd$Class &lt;- as.character(meltd$Class)

# calculate the total relative abundance for all taxa
means &lt;- ddply(meltd, ~Class, function(x) c(mean = mean(x$Abundance)))
means$mean &lt;- round(means$mean, digits = 8)
# this order in decending fashion
taxa_means &lt;- means[order(-means$mean), ]  
# ditch the sci notation 
taxa_means &lt;- format(taxa_means, scientific = FALSE)
#RENAME NA to UNCLASSIFIED
taxa_means$Class &lt;- gsub(&quot;NA&quot;, &quot;Unclassified&quot;, taxa_means$Class)</code></pre>
<p>Since our goal is to generate a figure and we only have 9 colors, some taxa will need to be put into an <strong>Other</strong> category. We can define ‘Other’ however we like so lets take a look at the overall relative abundance of each Class.</p>
<hr />
</div>
<div id="relative-abundance" class="section level3">
<h3>Relative abundance</h3>
<p><a id="relative abundance phyloseq object"></a></p>
<pre class="r"><code>datatable(taxa_means, rownames = FALSE, width = &quot;65%&quot;, 
          colnames = c(&quot;Class&quot;, &quot;mean&quot;), caption = 
            htmltools::tags$caption(style = &quot;caption-side: bottom; 
                                    text-align: left;&quot;, &quot;Table: &quot;,
                                    htmltools::em(&quot;Class-level 
                                                  relative abundance.&quot;)), 
          extensions = &quot;Buttons&quot;, 
          options = list(columnDefs = list(list(className = &quot;dt-center&quot;, 
                                                targets = &quot;_all&quot;)), 
                         dom = &quot;Blfrtip&quot;, pageLength = 10, 
                         lengthMenu = c(5, 10, 50, 70), 
                         buttons = c(&quot;csv&quot;, &quot;copy&quot;)))</code></pre>
<div id="htmlwidget-681a6addeee675513901" style="width:65%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-681a6addeee675513901">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; &#10;                                    text-align: left;\">\n  Table: \n  <em>Class-level \n                                                  relative abundance.<\/em>\n<\/caption>","data":[["Gammaproteobacteria","Alphaproteobacteria","Planctomycetacia","Clostridia","Fusobacteriia","Deltaproteobacteria","Bacteroidia","Oxyphotobacteria","Erysipelotrichia","Verrucomicrobiae","Spirochaetia","Mollicutes","Campylobacteria","Actinobacteria","Bacilli","Coriobacteriia","Acidimicrobiia","Dadabacteriia","Phycisphaerae","OM190","Lentisphaeria","Thermoanaerobaculia","Melainabacteria","vadinHA49","Anaerolineae","Saccharimonadia","Kiritimatiellae","Subgroup_9","Babeliae","Nitrososphaeria","Subgroup_22","Unclassified","Deferribacteres","Oligosphaeria","Subgroup_26","Deinococci","Chlorobia","KD4-96","Subgroup_17","Negativicutes","Zetaproteobacteria","Thermoleophilia","Rhodothermia","Pla3_lineage","Chlamydiae","BD7-11","Nitrospira","Holophagae","Halanaerobiia","Dehalococcoidia","Subgroup_6","Gracilibacteria","Acidobacteriia","Blastocatellia_(Subgroup_4)","KIST-JJY010","028H05-P-BN-P5","BD2-11_terrestrial_group","Fibrobacteria","Synergistia","Microgenomatia","Moduliflexia","PAUC43f_marine_benthic_group","Pla4_lineage","Entotheonellia","Hydrogenedentia","Ignavibacteria","Ktedonobacteria","Lineage_IIb","NC10","Calditrichia","WWE3","Parcubacteria"],["0.38183442","0.11797067","0.10258117","0.09487547","0.07117657","0.06379053","0.05990859","0.02762129","0.02703927","0.01374814","0.01136774","0.00913849","0.00235727","0.00217362","0.00154911","0.00140717","0.00120728","0.00118767","0.00117836","0.00109177","0.00077884","0.00040244","0.00028658","0.00027786","0.00025759","0.00023699","0.00022715","0.00022565","0.00021430","0.00018363","0.00016425","0.00014576","0.00013792","0.00012882","0.00010781","0.00007844","0.00006010","0.00005783","0.00005532","0.00005242","0.00005210","0.00005201","0.00005144","0.00003821","0.00002599","0.00002150","0.00001754","0.00001614","0.00001389","0.00001276","0.00001062","0.00000958","0.00000956","0.00000812","0.00000606","0.00000561","0.00000359","0.00000335","0.00000325","0.00000306","0.00000274","0.00000266","0.00000253","0.00000238","0.00000200","0.00000185","0.00000129","0.00000096","0.00000076","0.00000073","0.00000072","0.00000050"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Class<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":"_all"}],"dom":"Blfrtip","pageLength":10,"lengthMenu":[5,10,50,70],"buttons":["csv","copy"],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Inspecting the table it looks like if we choose a cutoff of 2% (0.02) we get 9 taxa—sounds pretty good. The rest go into the ‘Other’ category. No matter what, we will always gloss over some groups using such a coarse approach. But as we will see later, some of these low abundance groups will reappear when we look at the level of individual ASVs.</p>
<p>Here we define the <strong>Other</strong> category by combining all taxa with less than 2% of total reads.</p>
<pre class="r"><code>Other &lt;- means[means$mean &lt;= 0.02, ]$Class  
# or you can chose specifc taxa like this
# Other_manual &lt;- c(&quot;list&quot;, &quot;taxa&quot;, &quot;in&quot;, &quot;this&quot;, &quot;format&quot;)</code></pre>
<p>At a 2% abundance cutoff, 63 Classes are grouped into the ‘Other’ category. Next we will melt all these classes into the <strong>Other</strong> category and then craft the bar chart. It took some tweaking to get the bar chart to look just right—so there is a lot of code here—and it could most certainly be better. While we’re at it, we will also save a copy of the figure so we can tweak it later and make it look pretty.</p>
<pre class="r"><code>meltd[meltd$Class %in% Other, ]$Class &lt;- &quot;Other&quot;
samp_names &lt;- aggregate(meltd$Abundance, 
                        by = list(meltd$Sample), FUN = sum)[, 1]
.e &lt;- environment()
meltd[, &quot;Class&quot;] &lt;- factor(meltd[, &quot;Class&quot;], sort(unique(meltd[, &quot;Class&quot;])))
meltd &lt;- meltd[order(meltd[, &quot;Class&quot;]), ]
# Here we order Classes by the Phylum they belong to.
meltd$Class &lt;- factor(meltd$Class, 
                      levels = c(&quot;Bacteroidia&quot;, &quot;Clostridia&quot;, 
                                 &quot;Erysipelotrichia&quot;, &quot;Fusobacteriia&quot;, 
                                 &quot;Alphaproteobacteria&quot;, &quot;Deltaproteobacteria&quot;, 
                                 &quot;Gammaproteobacteria&quot;, &quot;Planctomycetacia&quot;, 
                                 &quot;Oxyphotobacteria&quot;, &quot;Other&quot;))  </code></pre>
<hr />
</div>
<div id="abundance-by-host-species" class="section level3">
<h3>Abundance by host species</h3>
<div class="paper">
Figure 2A
</div>
<pre class="r"><code>fig2A &lt;- ggplot(meltd, 
                aes_string(x = &quot;Sample&quot;, y = &quot;Abundance&quot;, fill = &quot;Class&quot;), 
                environment = .e, 
                ordered = TRUE, 
                xlab = &quot;x-axis label&quot;, ylab = &quot;y-axis label&quot;) 


fig2A &lt;- fig2A + geom_bar(stat = 
                            &quot;identity&quot;, 
                          position = position_stack(reverse = TRUE), 
                          width = 0.95) + 
  coord_flip() + 
  theme(aspect.ratio = 1 / 2)

fig2A &lt;- fig2A + scale_fill_manual(values = friend_pal)

fig2A &lt;- fig2A + theme(axis.text.x = element_text(angle = 0, 
                                                  hjust = 0.45, 
                                                  vjust = 1))

fig2A &lt;- fig2A + guides(fill = guide_legend(override.aes = list(colour = NULL), 
                                            reverse = FALSE)) + 
  theme(legend.key = element_rect(colour = &quot;black&quot;))

fig2A &lt;- fig2A + labs(x = &quot;Host species&quot;, 
                      y = &quot;Relative abundance (% total reads)&quot;, 
                      title = &quot;Abundance of bacterial taxa across host species&quot;)

fig2A &lt;- fig2A + theme(axis.line = element_line(colour = &quot;black&quot;), 
                       panel.grid.major = element_blank(), 
                       panel.grid.minor = element_blank(), 
                       panel.border = element_rect(colour = &quot;black&quot;, 
                                                   fill = NA, size = 1))

fig2A</code></pre>
<div class="figure" style="text-align: center">
<img src="4_diversity_files/figure-html/plot_bar_fig2A-1.png" alt="Figure 2A" width="672" />
<p class="caption">
Figure 2A
</p>
</div>
<pre class="r"><code>pdf(&quot;DATA/PHYLOSEQ/FIGURES/OUTPUT/Figure_2A.pdf&quot;)
fig2A
invisible(dev.off())</code></pre>
<hr />
<p>Armed with a picture of taxonomic composition we can move on to diversity estimates.</p>
</div>
</div>
<div id="alpha-diversity" class="section level2">
<h2>Alpha-Diversity</h2>
<div id="statistical-tests" class="section level3">
<h3>Statistical tests</h3>
<p>Alpha diversity describes the diversity in a sample or site. There are several alpha diversity metrics available in phyloseq: <code>Observed</code>, <code>Chao1</code>, <code>ACE</code>, <code>Shannon</code>, <code>Simpson</code>, <code>InvSimpson</code>, <code>Fisher</code>. Play around to see how different metrics change or confirm these results.</p>
<p>Here we want to know if diversity is significantly different across host species. In order to do that we need to know if we should run a parametric or non-parametric test, and for that we need to know if our data is normally distributed. Most of the ideas/code for alpha (and subsequent beta) diversity statistics come from this <a href="https://rpubs.com/maddieSC/R_SOP_UCR_Jan_2018" target="_blank">workshop tutorial</a> by Kim Dill-McFarland and Madison Cox.</p>
<p>First we run the diversity estimates, add these data to our summary table, and save a copy of this table.</p>
<div class="paper">
Table S3
</div>
<pre class="r"><code>diversity &lt;- estimate_richness(ps_slv_work_filt, 
                               measures = c(&quot;Observed&quot;, &quot;Chao1&quot;, &quot;ACE&quot;, 
                                          &quot;Shannon&quot;, &quot;Simpson&quot;, &quot;InvSimpson&quot;, 
                                          &quot;Fisher&quot;))

diversity_calc &lt;- diversity %&gt;% rownames_to_column(&quot;host_ID&quot;)
# round values
diversity_calc[c(3, 5, 10)] &lt;- round(diversity_calc[c(3, 5, 10)], 1)
diversity_calc[c(4, 6, 7, 9)] &lt;- round(diversity_calc[c(4, 6, 7, 9)], 2)
diversity_calc[8] &lt;- round(diversity_calc[8], 3)

host_summary &lt;- merge(host_details, diversity_calc)
host_summary$Observed &lt;- NULL
host_summary &lt;- host_summary[c(1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 13,
                               14, 15, 6, 7, 16, 17, 18, 19, 20, 21, 22, 23)]

write.table(host_summary, &quot;DATA/PHYLOSEQ/TABLES/OUTPUT/SUPP/Table_S3.txt&quot;, 
            sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE, 
            col.names = c(&quot;Sample ID&quot;, &quot;Host genus&quot;, &quot;Host species&quot;, 
                          &quot;Common name&quot;, &quot;NCBI tAxID&quot;, &quot;Collection date&quot;, 
                          &quot;Life phase&quot;, &quot;Weight (g)&quot;, &quot;Total length (cm)&quot;, 
                          &quot;Foregut length (cm)&quot;, &quot;Midgut length (cm)&quot;, 
                          &quot;Hindgut length (cm)&quot;, &quot;Total gut length (cm)&quot;, 
                          &quot;Total reads&quot;, &quot;Total ASVs&quot;, &quot;Chao1&quot;, &quot;Chao1 (se)&quot;, 
                          &quot;ACE&quot;, &quot;ACE (se)&quot;, &quot;Shannon&quot;, &quot;Simpson&quot;, 
                          &quot;InvSimpson&quot;, &quot;Fisher&quot;))

datatable(host_summary, rownames = FALSE, width = &quot;100%&quot;, 
          colnames = c(&quot;Sample ID&quot;, &quot;Host genus&quot;, &quot;Host species&quot;, 
                       &quot;Common name&quot;, &quot;NCBI tAxID&quot;, &quot;Collection date&quot;, 
                       &quot;Life phase&quot;, &quot;Weight (g)&quot;, &quot;Total length (cm)&quot;, 
                       &quot;Foregut length (cm)&quot;, &quot;Midgut length (cm)&quot;, 
                       &quot;Hindgut length (cm)&quot;, &quot;Total gut length (cm)&quot;, 
                       &quot;Total reads&quot;, &quot;Total ASVs&quot;, &quot;Chao1&quot;, &quot;Chao1 (se)&quot;, 
                       &quot;ACE&quot;, &quot;ACE (se)&quot;, &quot;Shannon&quot;, &quot;Simpson&quot;, &quot;InvSimpson&quot;, 
                       &quot;Fisher&quot;), 
            caption = 
            htmltools::tags$caption(style = 
                                      &quot;caption-side: bottom; text-align: 
                                    left;&quot;, &quot;Table: &quot;, 
                                    htmltools::em(&quot;Host-associated metadata &amp; 
                                                  microbial diversity&quot;)), 
          extensions = &quot;Buttons&quot;, options = 
            list(columnDefs = list(list(className = &quot;dt-left&quot;, targets = 0)), 
                 dom = &quot;Blfrtip&quot;, pageLength = 5, lengthMenu = c(5, 10, 50), 
                 buttons = c(&quot;csv&quot;, &quot;copy&quot;), scrollX = TRUE, 
                 scrollCollapse = TRUE))</code></pre>
<div id="htmlwidget-745e363e2cfa33de974f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-745e363e2cfa33de974f">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption style=\"caption-side: bottom; text-align: &#10;                                    left;\">\n  Table: \n  <em>Host-associated metadata &amp; \n                                                  microbial diversity<\/em>\n<\/caption>","data":[["AcCoe01","AcCoe02","AcCoe03","AcCoe04","AcCoe05","AcCoe06","AcCoe07","AcCoe08","AcTra01","AcTra02","AcTra03","AcTra04","AcTra05","AcTra06","AcTra07","AcTra08","AcTra09","ScTae01","ScTae02","ScTae03","ScTae04","ScTae05","ScTae06","ScTae07","ScTae08","ScTae09","SpAur01","SpAur02","SpAur03","SpAur04","SpAur05","SpAur06","SpAur07","SpAur08","SpAur09","SpAur10","SpAur11","SpAur12","SpAur13","SpVir01","SpVir02","SpVir03","SpVir04","SpVir05","SpVir06","SpVir07","SpVir08","SpVir09","SpVir10","SpVir11"],["Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Acanthurus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Scarus","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma","Sparisoma"],["coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","coeruleus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","tractus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","taeniopterus","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","aurofrenatum","viride","viride","viride","viride","viride","viride","viride","viride","viride","viride","viride"],["blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","blue tang surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","fiveband surgeonfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","princess parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","redband parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish","stoplight parrotfish"],["157585","157585","157585","157585","157585","157585","157585","157585","1316013","1316013","1316013","1316013","1316013","1316013","1316013","1316013","1316013","544418","544418","544418","544418","544418","544418","544418","544418","544418","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59663","59666","59666","59666","59666","59666","59666","59666","59666","59666","59666","59666"],["2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-20","2016-7-20","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-20","2016-7-20","2016-7-20","2016-7-20","2016-7-20","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-4-24","2016-4-24","2016-4-24","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-15","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19","2016-7-19"],["large","large","large","large","large","large","large","juvenile","large","small","large","large","large","large","small","small","large","terminal","initial","terminal","terminal","terminal","terminal","initial","initial","initial","terminal","initial","terminal","initial","initial","initial","initial","terminal","initial","initial","juvenile","terminal","initial","initial","terminal","terminal","initial","initial","initial","initial","initial","initial","initial","initial"],[266,187.8,129.9,188.1,277.5,235.8,226,8,89.8,69,101.5,88.5,98,84,13,23,96,526,115,331,574,394.6,445.8,220.9,142.04,195.2,236,112,212,114,null,null,null,185.1,152.6,175.05,29.4,173.09,142.5,358,null,1124,408.41,564,552.8,310.1,300.45,645,243.5,150.5],[23.5,20,17.5,20.5,23,22.5,22,7,17.2,15,17,16.5,18.4,16,8.3,9.8,17.5,30.5,19,25.4,30.3,27.9,28.7,22,19.6,21.5,24.5,18,23.1,18.5,null,null,null,21.6,19.5,21.8,11.5,21,20,27,40,43,28.5,34,33.8,25.5,26,38,24.2,19.5],[17,9,7,9,15,10,14,4,7,10,10,null,4,null,3.5,5,5,11.5,7,8,19,10.5,10,9,4,8.5,11,9,7,5,null,null,null,8.8,9,10.5,3.1,7.2,6,10,16,null,11,null,12.5,12.8,10.7,13,6,null],[64,74,71,76,97,83,71,22,53,43,32,null,49,null,24,29,53,31.5,20,25,39,31.5,34.1,29,22,27,18,17,26,17,null,null,null,25.8,26.1,28.2,8.7,27.8,20.5,39,78,null,39,null,53,46,42.4,65.5,34,null],[9,12,4,7,7,5,7,4.5,6,3,4,null,3,null,3,4,4,5,1.5,3,4,5,3.5,2.6,3,4,4,2,3.5,5,null,null,null,3,3,3,0.7,2.6,3,3.5,10,null,5,null,4.8,2.5,4.3,4.7,3,null],[90,95,82,92,119,98,92,30.5,66,56,46,null,56,null,30.5,38,62,48,28.5,36,62,47,47.6,40.6,29,39.5,33,28,36.5,27,null,null,null,37.6,38.1,41.7,12.5,37.6,29.5,52.5,104,null,55,null,70.3,61.3,57.4,83.2,43,null],[64360,137180,105966,180159,58982,105859,78401,117863,31483,53448,47620,65285,88469,60784,48607,131146,57571,52898,36997,28642,80074,58809,53118,82019,42598,27312,42513,11686,43164,15359,45483,19398,37006,13220,28047,71679,26893,62031,87183,39112,71391,75561,19366,73304,20724,49961,44363,41082,74298,26522],[1148,630,702,461,827,789,475,402,534,194,793,1503,2031,1768,488,391,1516,795,771,676,1308,723,576,1022,297,786,175,154,153,332,187,612,257,96,486,592,198,915,350,470,1037,1548,418,598,666,594,665,634,572,108],[1175,642.5,704,469.2,841,795.8,476.3,405.9,552.2,197.5,815.2,1543.4,2090.7,1821.7,493,402,1544.6,813.3,801,704.6,1345.3,733,588.8,1036.5,297.9,809.3,175,154.2,153,334.4,188.2,628,264.6,101,500.6,601.1,198.7,929.6,357.3,472.3,1050.9,1604.4,422.8,604.4,672.2,600.5,684.8,639.4,575,109.5],[10.74,6.58,1.96,6.36,7.07,4.11,1.56,3.19,8.41,3.44,9.64,12.82,15.47,14.72,3.51,8.88,10.32,7.62,11.76,12.57,12.29,5.32,7.27,6.39,1.3,10.25,0.03,0.62,0.06,2.27,1.52,7.45,5.45,6.04,7.49,5.03,1.15,6.94,4.53,2.42,6.56,18.28,3.61,4.19,3.74,4.34,9.52,3.74,2.52,2.59],[1159.5,637.2,704.4,463.6,833.6,794.7,477.3,405.2,545.2,196.1,802.8,1523.5,2067.1,1799,492.8,393.1,1530.2,811.9,786.9,687.5,1332.3,731.2,581.4,1033.8,298.4,796.2,175.2,154.8,153.2,334.8,189.1,623.2,262.1,97.5,495.8,599.4,198.8,922.7,357,471.6,1046,1567.2,422,602.8,672.5,598.3,673.2,638.2,574.9,108.7],[16.74,12.43,12.84,10.36,14.26,13.89,9.96,9.9,11.49,6.88,13.97,18.86,21.73,20.66,11.01,9.8,19.02,13.65,13.55,12.76,17.15,13.14,11.94,15.67,8.59,13.79,6.57,6.11,6.13,9.01,6.43,12.19,7.85,4.55,10.81,12.09,6.96,15.03,9.33,10.59,15.91,18.94,10.18,12.09,12.79,12.05,12.79,12.38,11.92,4.63],[5.1,3.51,4.39,2.94,4,4,4.56,3.04,3.95,2.92,4.52,6.09,6.49,6.45,3.52,1.95,6.27,4.98,4.98,4.67,5.81,4.1,3.23,3.98,2.51,5.48,2.4,2.2,2.7,3.44,3.51,4.89,2.8,2.81,4.21,2.73,1.31,3.94,2.32,3.76,4.78,6.04,4.33,3.77,5.3,3.95,4.49,4.29,3.47,2.85],[0.968,0.927,0.973,0.821,0.866,0.947,0.971,0.882,0.912,0.882,0.955,0.993,0.995,0.995,0.898,0.577,0.993,0.962,0.963,0.937,0.987,0.924,0.838,0.845,0.776,0.983,0.792,0.625,0.854,0.849,0.929,0.942,0.805,0.878,0.952,0.725,0.354,0.852,0.714,0.887,0.972,0.987,0.964,0.942,0.979,0.929,0.97,0.943,0.932,0.877],[31.07,13.76,36.89,5.57,7.44,19.03,34.32,8.48,11.4,8.5,22.11,143.66,187.66,184.41,9.76,2.36,147.97,26.23,26.7,15.98,74.68,13.11,6.18,6.47,4.46,58.04,4.8,2.67,6.87,6.64,14.01,17.37,5.14,8.19,20.78,3.64,1.55,6.74,3.49,8.85,35.25,76.44,27.68,17.34,48.42,14.04,33.78,17.66,14.78,8.15],[198.5,85.3,100.9,57.2,136.2,115.7,67.3,52,91.4,25.3,135.2,274.5,370.7,340.7,75.4,49.6,285.4,132.7,137.8,124.1,222,116.1,90.3,164.5,43.1,151.1,23.3,25.1,19.9,59.8,24.9,120.2,37.2,14,83.5,88.4,29,152.2,46.4,75.1,171.9,275.6,75.3,89.1,131.4,94.7,111,106.4,84.3,14.4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample ID<\/th>\n      <th>Host genus<\/th>\n      <th>Host species<\/th>\n      <th>Common name<\/th>\n      <th>NCBI tAxID<\/th>\n      <th>Collection date<\/th>\n      <th>Life phase<\/th>\n      <th>Weight (g)<\/th>\n      <th>Total length (cm)<\/th>\n      <th>Foregut length (cm)<\/th>\n      <th>Midgut length (cm)<\/th>\n      <th>Hindgut length (cm)<\/th>\n      <th>Total gut length (cm)<\/th>\n      <th>Total reads<\/th>\n      <th>Total ASVs<\/th>\n      <th>Chao1<\/th>\n      <th>Chao1 (se)<\/th>\n      <th>ACE<\/th>\n      <th>ACE (se)<\/th>\n      <th>Shannon<\/th>\n      <th>Simpson<\/th>\n      <th>InvSimpson<\/th>\n      <th>Fisher<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-left","targets":0},{"className":"dt-right","targets":[7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22]}],"dom":"Blfrtip","pageLength":5,"lengthMenu":[5,10,50],"buttons":["csv","copy"],"scrollX":true,"scrollCollapse":true,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="warning">
<p>This table also scrolls horizontally.</p>
</div>
<hr />
<p>Next, we add the diversity estimates to our phyloseq object, and test if the data are normally distributed using Shapiro-Wilk Normality test. We will focus on the inverse Simpson and Shannon diversity estimates and Chao’s richness estimate but this approach can be used for any metric.</p>
<pre class="r"><code># Convert to ps object
sample_div &lt;- sample_data(diversity) 
# Create new ps object with diversity estimates added to sample_data
ps_slv_work_filt_div &lt;- merge_phyloseq(ps_slv_work_filt, sample_div)
# Run Shapiro test
shapiro_test_Shan &lt;- shapiro.test(sample_data(ps_slv_work_filt_div)$Shannon)
shapiro_test_invSimp &lt;- shapiro.test(sample_data(ps_slv_work_filt_div)$InvSimpson)
shapiro_test_Chao1 &lt;- shapiro.test(sample_data(ps_slv_work_filt_div)$Chao1)
shapiro_test_Observed &lt;- shapiro.test(sample_data(ps_slv_work_filt_div)$Observed)</code></pre>
<p>Shapiro-Wilk Normality Test for <strong>Shannon</strong> index.</p>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  sample_data(ps_slv_work_filt_div)$Shannon
## W = 0.98228, p-value = 0.6513</code></pre>
<p>Shapiro-Wilk Normality Test for <strong>inverse Simpson</strong> index.</p>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  sample_data(ps_slv_work_filt_div)$InvSimpson
## W = 0.60454, p-value = 0.0000000002276</code></pre>
<p>Shapiro-Wilk Normality Test for <strong>Chao1 richness</strong> estimator.</p>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  sample_data(ps_slv_work_filt_div)$Chao1
## W = 0.89305, p-value = 0.0002855</code></pre>
<p>Shapiro-Wilk Normality Test for <strong>Observed ASV richness</strong> estimator.</p>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  sample_data(ps_slv_work_filt_div)$Observed
## W = 0.89591, p-value = 0.0003531</code></pre>
<p>Ok, since the p-values are significant for the inverse Simpson, Chao richness, and Observed ASV richness we reject the null hypothesis that these data are normally distributed. However, the Shannon estimates appear normally distributed. So lets see if diversity is significantly different between host species based on the Shannon index.</p>
<div id="normally-distributed" class="section level4">
<h4>Normally distributed</h4>
<p>Since the Shannon data is normally distributed we can test for significance using ANOVA (a parametric test).</p>
<pre class="r"><code>sampledataDF &lt;- data.frame(sample_data(ps_slv_work_filt_div))
aov.shannon &lt;- aov(Shannon ~ Sp, data = sampledataDF)
#Call for the summary of that ANOVA, which will include P-values
summary(aov.shannon)</code></pre>
<pre><code>##             Df Sum Sq Mean Sq F value  Pr(&gt;F)   
## Sp           4  19.13   4.782   3.906 0.00833 **
## Residuals   45  55.10   1.224                   
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Ok, the results of the ANOVA are significant. Here we use the Tukey’s HSD (honestly significant difference) post-hoc test to determine which pairwise comparisons are different.</p>
<pre class="r"><code>TukeyHSD(aov.shannon)</code></pre>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = Shannon ~ Sp, data = sampledataDF)
## 
## $Sp
##                   diff         lwr         upr     p adj
## AcTra-AcCoe  0.7421379 -0.78560423  2.26987998 0.6431490
## ScTae-AcCoe  0.4727303 -1.05501185  2.00047236 0.9030495
## SpAur-AcCoe -0.9230990 -2.33591246  0.48971439 0.3551484
## SpVir-AcCoe  0.3323910 -1.12853187  1.79331396 0.9664219
## ScTae-AcTra -0.2694076 -1.75153517  1.21271993 0.9852679
## SpAur-AcTra -1.6652369 -3.02859596 -0.30187785 0.0096995
## SpVir-AcTra -0.4097468 -1.82289999  1.00340634 0.9218562
## SpAur-ScTae -1.3958293 -2.75918834 -0.03247023 0.0424253
## SpVir-ScTae -0.1403392 -1.55349237  1.27281396 0.9985589
## SpVir-SpAur  1.2554901 -0.03255018  2.54353034 0.0593081</code></pre>
<p>Looks like <em>Sparisoma aurofrenatum</em> is significantly different from <em>Scarus taeniopterus</em> and <em>Acanthurus tractus</em>.</p>
</div>
<div id="non-normally-distributed" class="section level4">
<h4>Non-normally distributed</h4>
<p>Now we can look at the results on the inverse Simpson diversity and Chao’s richness. Since host species is categorical, we use Kruskal-Wallis (non-parametric equivalent of ANOVA) to test for significance.</p>
<hr />
<p>Kruskal-Wallis of <strong>inverse Simpson</strong> index.</p>
<pre class="r"><code>#library(FSA)
#dunnTest(InvSimpson ~ Sp, data = sampledataDF, method=&quot;bh&quot;) 
kruskal.test(InvSimpson ~ Sp, data = sampledataDF)</code></pre>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  InvSimpson by Sp
## Kruskal-Wallis chi-squared = 13.779, df = 4, p-value = 0.008034</code></pre>
<p>Kruskal-Wallis of <strong>Chao1 richness</strong> estimator.</p>
<pre class="r"><code>#dunnTest(Chao1 ~ Sp, data = sampledataDF, method=&quot;bh&quot;) 
kruskal.test(Chao1 ~ Sp, data = sampledataDF)</code></pre>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  Chao1 by Sp
## Kruskal-Wallis chi-squared = 13.992, df = 4, p-value = 0.007321</code></pre>
<p>Kruskal-Wallis of <strong>Observed ASV richness</strong> index.</p>
<pre class="r"><code>#library(FSA)
#dunnTest(Observed ~ Sp, data = sampledataDF, method=&quot;bh&quot;) 
kruskal.test(Observed ~ Sp, data = sampledataDF)</code></pre>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  Observed by Sp
## Kruskal-Wallis chi-squared = 14.153, df = 4, p-value = 0.006822</code></pre>
<hr />
<p>For the inverse Simpson, Chao1, and Observed richness the results of the Kruskal-Wallis rank sum test are significant. So we can look at pairwise comparisons using Wilcoxon rank sum test for post-hoc analysis.</p>
<p>Pairwise significance test for <strong>inverse Simpson</strong> index.</p>
<pre class="r"><code>pairwise.wilcox.test(sampledataDF$InvSimpson, sampledataDF$Sp, 
                     p.adjust.method = &quot;fdr&quot;)</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum test 
## 
## data:  sampledataDF$InvSimpson and sampledataDF$Sp 
## 
##       AcCoe AcTra ScTae SpAur
## AcTra 0.545 -     -     -    
## ScTae 0.963 0.545 -     -    
## SpAur 0.042 0.042 0.108 -    
## SpVir 0.545 0.789 0.545 0.004
## 
## P value adjustment method: fdr</code></pre>
<p>Pairwise significance test for <strong>Chao1 richness</strong> estimator.</p>
<pre class="r"><code>pairwise.wilcox.test(sampledataDF$Chao1, sampledataDF$Sp, 
                     p.adjust.method = &quot;fdr&quot;)</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum test 
## 
## data:  sampledataDF$Chao1 and sampledataDF$Sp 
## 
##       AcCoe AcTra ScTae SpAur
## AcTra 0.628 -     -     -    
## ScTae 0.617 0.666 -     -    
## SpAur 0.030 0.030 0.019 -    
## SpVir 0.666 0.628 0.304 0.046
## 
## P value adjustment method: fdr</code></pre>
<p>Pairwise significance test for <strong>Observed ASV richness</strong> index.</p>
<pre class="r"><code>pairwise.wilcox.test(sampledataDF$Observed, sampledataDF$Sp, 
                     p.adjust.method = &quot;fdr&quot;)</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum test 
## 
## data:  sampledataDF$Observed and sampledataDF$Sp 
## 
##       AcCoe AcTra ScTae SpAur
## AcTra 0.677 -     -     -    
## ScTae 0.677 0.730 -     -    
## SpAur 0.026 0.026 0.019 -    
## SpVir 0.730 0.677 0.304 0.039
## 
## P value adjustment method: fdr</code></pre>
<p>Again we see that only <em>Sp. aurofrenatum</em> is significantly different from the other hosts. For the inverse Simpson index, <em>Sp. aurofrenatum</em> is significantly different from three of the four host species and Chao1 richness estimator, <em>Sp. aurofrenatum</em> is significantly different from all other host species. Now we can plot the results.</p>
<hr />
</div>
</div>
<div id="diversity-plots" class="section level3">
<h3>Diversity plots</h3>
<p>Here we plot the results of Shannon diversity index. We will save a copy of the figure for later tweaking. We use the color palette described above to delineate host species.</p>
<div class="paper">
Figure 2B
</div>
<pre class="r"><code>fig2B &lt;- plot_richness(ps_slv_work_filt, x = &quot;Sp&quot;, 
                       measures = c(&quot;Observed&quot;,
                                    &quot;Shannon&quot;, 
                                    &quot;InvSimpson&quot;, 
                                    &quot;Chao1&quot;), 
                       color = &quot;Sp&quot;, nrow = 1)
fig2B &lt;- fig2B + geom_boxplot() + geom_jitter(width = 0.05)
fig2B &lt;- fig2B + scale_colour_manual(values = samp_pal) + 
         labs(x = &quot;Host species&quot;, 
         y = &quot;Diversity&quot;, 
         title = &quot;Alpha diversity of bacterial 
         communities in herbivorous reef fish&quot;)
#fig2B + geom_boxplot(aes(colour = black))
#fig2B &lt;- fig2B + theme_bw() + geom_point(size = 2.5, aes(color = Sp)) + 
fig2B</code></pre>
<div class="figure" style="text-align: center">
<img src="4_diversity_files/figure-html/alpha_div_fig_2B-1.png" alt="Figure 2B" width="672" />
<p class="caption">
Figure 2B
</p>
</div>
<pre class="r"><code>pdf(&quot;DATA/PHYLOSEQ/FIGURES/OUTPUT/Figure_2B.pdf&quot;)
fig2B
invisible(invisible(dev.off()))</code></pre>
<hr />
</div>
<div id="correlations-with-diversity" class="section level3">
<h3>Correlations with diversity</h3>
<p>Next we wanted to know if any alpha-diversity metrics were correlated with host physical characteristics. At the time of collection, we recorded host weight, total length, total gut length, as well as the length of individual gut segments (fore, mid, hind).</p>
<p>When considering the dataset as a whole (i.e., all samples), we found no correlation between any physical characteristics and any diversity metrics. If we split samples by genera we found that neither <em>Acanthurus</em> nor <em>Scarus</em> were not significant for any parameters while <em>Sparisoma</em> showed significant results for all parameters except hindgut_length.</p>
<pre class="r"><code>dt &lt;- read.table(&quot;DATA/PHYLOSEQ/TABLES/OUTPUT/SUPP/Table_S3.txt&quot;, 
                 sep = &quot;\t&quot;, header = TRUE)
library(ggpubr)
scarus &lt;- host_summary[host_summary$host_genus %in% &quot;Scarus&quot;, ]
sparisoma &lt;- host_summary[host_summary$host_genus %in% &quot;Sparisoma&quot;, ]
acanthurus &lt;- host_summary[host_summary$host_genus %in% &quot;Acanthurus&quot;, ]
alphametric &lt;- c(&quot;total_ASVs&quot;, &quot;Chao1&quot;, &quot;ACE&quot;, &quot;Shannon&quot;, 
                &quot;Simpson&quot;,  &quot;InvSimpson&quot;, &quot;Fisher&quot;)
physical_char &lt;- c(&quot;weight&quot;, &quot;total_length&quot;, &quot;foregut_length&quot;, 
                   &quot;midgut_length&quot;, &quot;hindgut_length&quot;, &quot;total_gut_length&quot;)

# Full set: not significant -&gt; &quot;weight&quot;, &quot;total_length&quot;
            #&quot;foregut_length&quot;, R = 0.32
            #&quot;midgut_length&quot;, R = 0.50
            #&quot;hindgut_length&quot;, R = 0.17-0.34
            #&quot;total_gut_length&quot; R = 0.50

#By genus and midgut_length :
          # scarus NS
          # sparisoma R = 0.8
          # acanthurus NS

# acanthurus not significant for any parameters
# scarus not significant for any parameters
# sparisoma  significant for all parameters except hindgut_length was a bit weak
# &quot;weight&quot;, &quot;total_length&quot; &quot;foregut_length&quot; &quot;midgut_length&quot; 
# &quot;hindgut_length&quot; &quot;total_gut_length&quot;

# To do all diversity metric  change &quot;y = &quot; to y = alphametric and
# ylab = alphametric

par(mfrow = c(2, 3))
shan_by_length &lt;- ggscatter(host_summary, x = &quot;total_gut_length&quot;, 
                            y = &quot;Shannon&quot;, add = &quot;reg.line&quot;, 
                            conf.int = FALSE,cor.coef = TRUE, 
                            cor.method = &quot;spearman&quot;,
                            xlab = &quot;total_length (cm)&quot;, ylab = &quot;Shannon&quot;, 
                            color = &quot;host_genus&quot;, palette = samp_pal,
                            legend = &quot;bottom&quot;)

shan_by_weight &lt;- ggscatter(host_summary, x = &quot;weight&quot;, y = &quot;Shannon&quot;, 
          add = &quot;reg.line&quot;, conf.int = FALSE, 
          cor.coef = TRUE, cor.method = &quot;spearman&quot;,
          xlab = &quot;weight (g)&quot;, ylab = &quot;Shannon&quot;, 
          color = &quot;host_genus&quot;, palette = samp_pal, legend = &quot;top&quot;)

grid.arrange(shan_by_length, shan_by_weight, ncol = 2)</code></pre>
<p><img src="4_diversity_files/figure-html/run_correlations-1.png" width="672" /></p>
<pre class="r"><code>#ggscatter(host_summary, x = &quot;total_gut_length&quot;, y = alphametric, 
#          add = &quot;reg.line&quot;, conf.int = FALSE, 
#          cor.coef = TRUE, cor.method = &quot;spearman&quot;,
#          xlab = &quot;mid-gut length (cm)&quot;, ylab = alphametric, 
#          color = &quot;host_genus&quot;, palette = samp_pal)


#, facet.by = &quot;host_species&quot;)
######################################################
#shapiro.test(host_summary$Shannon) # =&gt; p = 0.1229
## Shapiro-Wilk normality test for wt
#shapiro.test(host_summary$midgut_length) # =&gt; p = 0.09
#ggqqplot(host_summary$Shannon) # =&gt; p = 0.1229
## Shapiro-Wilk normality test for wt
#ggqqplot(host_summary$midgut_length) # =&gt; p = 0.09
######################################################</code></pre>
<!--TODO: change boxes and lines to black--add point boarders-->
</div>
</div>
<div id="beta-diversity" class="section level2">
<h2>Beta-Diversity</h2>
<p>Beta diversity basically tells us how similar or dissimilar samples are to one another. Phyloseq offers several ordination <code>methods</code> and <code>distance</code> metrics. Here we use non metric multidimensional scaling (NMDS) coupled with Jensen–Shannon divergence. We also save a copy of the figure for later tweaking.</p>
<div class="warning">
<p>To see the full output of the NMDS analysis, remove the <code>results = 'hide'</code> tag from the code chunk.</p>
</div>
<pre class="r"><code>set.seed(3131)
ord.nmds.jsd_slv &lt;- ordinate(ps_slv_work_filt, method = &quot;NMDS&quot;, 
                             distance = &quot;jsd&quot;)
stressplot(ord.nmds.jsd_slv)</code></pre>
<p><img src="4_diversity_files/figure-html/run_nmds-1.png" width="672" /></p>
<pre><code>## 
## Call:
## metaMDS(comm = ps.dist) 
## 
## global Multidimensional Scaling using monoMDS
## 
## Data:     ps.dist 
## Distance: user supplied 
## 
## Dimensions: 2 
## Stress:     0.1646318 
## Stress type 1, weak ties
## Two convergent solutions found after 20 tries
## Scaling: centring, PC rotation 
## Species: scores missing</code></pre>
<p>We see that a convergent solution was reached around 20 iterations and our stress is below 0.20, meaning that 2-axes are sufficient to view the data. Generally, we are looking for stress values below 0.2. If the stress values are high, you may need to add more axes to the ordination. Lets visualize the plot.</p>
<div id="diversity-plots-1" class="section level3">
<h3>Diversity plots</h3>
<div class="paper">
Figure 2C
</div>
<pre class="r"><code>fig2C &lt;- plot_ordination(ps_slv_work_filt, ord.nmds.jsd_slv, 
                         color = &quot;Sp&quot;, label = &quot;SamName&quot;, 
                         title = &quot;Jensen-Shannon divergence&quot;)

fig2C &lt;- fig2C + geom_point(size = 4) + 
  geom_point(shape = 1, size = 3.6, colour = &quot;black&quot;, stroke = 0.75) 
# +  xlim(-0.4, 0.4) + ylim(-0.4, 0.4)

fig2C &lt;- fig2C + scale_colour_manual(values = samp_pal)

fig2C &lt;- fig2C + theme(axis.line = element_line(colour = &quot;black&quot;), 
                       panel.background = element_blank(), 
                       panel.grid.major = element_line(&quot;grey&quot;), 
                       panel.grid.minor = element_line(&quot;grey&quot;), 
                       panel.border = 
                         element_rect(colour = &quot;black&quot;, fill = NA, size = 1)) + 
  theme(legend.key = element_rect(colour = &quot;black&quot;))

fig2C &lt;- fig2C + coord_fixed()
fig2C &lt;- fig2C  + stat_ellipse(type = &quot;t&quot;) + theme_bw()
fig2C</code></pre>
<div class="figure" style="text-align: center">
<img src="4_diversity_files/figure-html/beta_div_fig_2C-1.png" alt="Figure 2C" width="672" />
<p class="caption">
Figure 2C
</p>
</div>
<pre class="r"><code>pdf(&quot;DATA/PHYLOSEQ/FIGURES/OUTPUT/Figure_2C.pdf&quot;)
fig2C
invisible(dev.off())</code></pre>
<p>So we can see some clustering within groups and spread between groups, but this is not a test for statistical differences. Do microbial communities differ significantly by host taxa?</p>
</div>
<div id="permanova" class="section level3">
<h3>PERMANOVA</h3>
<p>To test whether microbial communities differ by host species we can use permutational analysis of variance (PERMANOVA) or analysis of similarity (ANOSIM). PERMANOVA does not assume normality but does assume equal beta dispersion between groups. We will test beta dispersion below.</p>
<p>First we use the <code>adonis</code> function in vegan to run a PERMANOVA test. This will tell us whether host species have similar centroids or not.</p>
<pre class="r"><code>set.seed(1911)
fish.jsd &lt;- phyloseq::distance(ps_slv_work_filt, method = &quot;jsd&quot;)
sampledf &lt;- data.frame(sample_data(ps_slv_work_filt))
fish_adonis &lt;- adonis(fish.jsd ~ Sp, data = sampledf, permutations = 1000)
fish_adonis</code></pre>
<pre><code>## 
## Call:
## adonis(formula = fish.jsd ~ Sp, data = sampledf, permutations = 1000) 
## 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2   Pr(&gt;F)    
## Sp         4    4.4045 1.10113   16.52 0.59489 0.000999 ***
## Residuals 45    2.9995 0.06665         0.40511             
## Total     49    7.4040                 1.00000             
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>These results indicate that centroids are significantly different across host species meaning that communities are different by host species.</p>
<p>We can also use the <code>pairwiseAdonis</code> package for pair-wise PERMANOVA analysis.</p>
<pre class="r"><code>pairwise.adonis(fish.jsd, factors = sampledf$Sp, p.adjust.m = &quot;bonferroni&quot;)</code></pre>
<pre><code>##             pairs Df SumsOfSqs   F.Model        R2 p.value p.adjusted sig
## 1  AcCoe vs AcTra  1 0.9511904  9.643336 0.3913162   0.002       0.02   .
## 2  AcCoe vs ScTae  1 1.3507311 21.403824 0.5879554   0.001       0.01   *
## 3  AcCoe vs SpAur  1 1.5267419 25.208974 0.5702230   0.001       0.01   *
## 4  AcCoe vs SpVir  1 1.2852392 16.065426 0.4858678   0.001       0.01   *
## 5  AcTra vs ScTae  1 0.7586873 10.699312 0.4007336   0.001       0.01   *
## 6  AcTra vs SpAur  1 1.3229620 19.765485 0.4970513   0.001       0.01   *
## 7  AcTra vs SpVir  1 0.9806215 11.402876 0.3878150   0.001       0.01   *
## 8  ScTae vs SpAur  1 1.0351877 25.696356 0.5623283   0.001       0.01   *
## 9  ScTae vs SpVir  1 1.0661860 18.907582 0.5122953   0.001       0.01   *
## 10 SpAur vs SpVir  1 0.7558760 13.640523 0.3827251   0.001       0.01   *</code></pre>
<p>Here we see again we see that communities are different by host species.</p>
<p>However, PERMANOVA assumes equal beta dispersion so we will use the <code>betadisper</code> function from the <code>vegan</code> package to calculate beta dispersion values.</p>
<pre class="r"><code>beta_adonis &lt;- betadisper(fish.jsd, sampledf$Sp, bias.adjust = TRUE)
beta_adonis</code></pre>
<pre><code>## 
##  Homogeneity of multivariate dispersions
## 
## Call: betadisper(d = fish.jsd, group = sampledf$Sp, bias.adjust =
## TRUE)
## 
## No. of Positive Eigenvalues: 34
## No. of Negative Eigenvalues: 15
## 
## Average distance to median:
##  AcCoe  AcTra  ScTae  SpAur  SpVir 
## 0.2980 0.3098 0.1845 0.1816 0.2570 
## 
## Eigenvalues for PCoA axes:
## (Showing 8 of 49 eigenvalues)
##  PCoA1  PCoA2  PCoA3  PCoA4  PCoA5  PCoA6  PCoA7  PCoA8 
## 1.8547 1.5977 0.9602 0.7234 0.5659 0.3413 0.2609 0.2427</code></pre>
<p>And then a pair-wise Permutation test for homogeneity of multivariate dispersions using <code>permutest</code> (again from the <code>vegan</code> package).</p>
<pre class="r"><code>permutest(beta_adonis, pairwise = TRUE, permutations = 1000)</code></pre>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 1000
## 
## Response: Distances
##           Df  Sum Sq  Mean Sq      F N.Perm   Pr(&gt;F)   
## Groups     4 0.14600 0.036500 4.1891   1000 0.007992 **
## Residuals 45 0.39209 0.008713                          
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##           AcCoe     AcTra     ScTae     SpAur  SpVir
## AcCoe           0.8051948 0.0029970 0.0119880 0.2797
## AcTra 0.8147277           0.0159840 0.0139860 0.2557
## ScTae 0.0040024 0.0160930           0.9440559 0.0420
## SpAur 0.0125449 0.0160815 0.9425028           0.0699
## SpVir 0.2735320 0.2661694 0.0479863 0.0669727</code></pre>
<p>These results are significant, meaning that host species have different dispersions. Looking at the pairwise p-values and permuted p-value, we see that the significant differences (p-value &lt; 0.05) are between:</p>
<ul>
<li>SpAur &amp; AcCoe, AcTra</li>
<li>ScTae &amp; AcCoe, AcTra, SpVir</li>
</ul>
<p>This means we are less confident that the PERMANOVA result is a real result, and that the result is possibly due to differences in group dispersions.</p>
</div>
<div id="anosim" class="section level3">
<h3>ANOSIM</h3>
<p>We can also use Analysis of Similarity (ANOSIM)—which does not assume equal group variances—to test whether overall microbial communities differ by host species.</p>
<pre class="r"><code>spgroup &lt;- get_variable(ps_slv_work_filt, &quot;Sp&quot;)
fish_anosim &lt;- anosim(distance(ps_slv_work_filt, &quot;jsd&quot;), grouping = spgroup)
summary(fish_anosim)</code></pre>
<pre><code>## 
## Call:
## anosim(x = distance(ps_slv_work_filt, &quot;jsd&quot;), grouping = spgroup) 
## Dissimilarity: 
## 
## ANOSIM statistic R: 0.8544 
##       Significance: 0.001 
## 
## Permutation: free
## Number of permutations: 999
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0556 0.0758 0.0873 0.1111 
## 
## Dissimilarity ranks between and within classes:
##         0%    25%   50%    75% 100%   N
## Between 94 462.50 726.5 977.25 1225 992
## AcCoe   36 185.50 290.5 360.00  582  28
## AcTra   14 162.25 322.0 510.50  789  36
## ScTae    9  34.75  74.5 122.00  419  36
## SpAur    1  30.50  76.0 137.50  562  78
## SpVir   13  95.00 177.0 273.00  584  55</code></pre>
<p>And the AN0SIM result is significant meaning that host species influences microbial community composition.</p>
<p></br></p>
<div style="text-align:center">
<h4>
<button class="favorite styled" type="button">
<a href="5_da_asv.html">Go to Part 5, Differentially Abundant ASV</a>
</button>
</h4>
</div>
</div>
</div>

<footer role="contentinfo" id="site-footer">

<hr>

<!-- /.bottom-menu -->
	<p class="copyright">&#169; 2019 metacrobe & ProjectDIGEST,
		powered by <a href="http://www.r-project.org">R</a> + <a href="http://rmarkdown.rstudio.com">RMarkdown</a> + <a href="http://github.com">Github</a>.</p>
</footer>

<!--this prevents a white space at bottom of page -->
<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;"></div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
