<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Appendices</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DIGEST</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="0_pw_intro.html">
    <span class="fa fa-info"></span>
     
    Intro
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-wrench"></span>
     
    Workflow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="1_field_observations.html">1. Field Observations</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Processing</li>
    <li>
      <a href="2_dada2.html">2. DADA2</a>
    </li>
    <li>
      <a href="3_data_prep.html">3. Data Prep</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">16S rRNA Analyses</li>
    <li>
      <a href="4_diversity.html">4. Diversity</a>
    </li>
    <li>
      <a href="5_da_asv.html">5. DA ASVs</a>
    </li>
    <li>
      <a href="6_synthesis.html">6. Synthesis</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="7_appendices.html">Appendices</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="data_availability.html">
    <span class="fa fa-database"></span>
     
    Data
  </a>
</li>
<li>
  <a href="supplemental_material.html">SOM</a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-phone"></span>
     
    Contact
  </a>
</li>
<li>
  <a href="how_to_build_this_site.html">
    <span class="fa fa-question fa-lg"></span>
     
    About
  </a>
</li>
<li>
  <a href="https://github.com/projectdigest/web/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="raw_code.txt">
    <span class="fa fa-code fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="people.html">
    <span class="fa fa-user-friends"></span>
     
    People
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Appendices</h1>

</div>


<p><a href="0_pw_intro.html#workflow_overview">Back to Introduction</a></p>
<hr />
<div id="appendix-a-other-analyses-visualizations" class="section level2">
<h2>Appendix A: Other analyses &amp; visualizations</h2>
<p>Here is code for other representation of taxa abundance. These are raw <code>R</code> images that have not been gussied up.</p>
<p>We will create two different representations of relative abundance for each sample (arranged by host species) by major Classes—facet grid box-and-whisker plots and bar charts. We will generate each separately (and save) using an earlier relative abundance phyloseq object and then display the combined output.</p>
<div id="code-for-box-and-whisker-plot" class="section level4">
<h4>Code for box-and-whisker plot</h4>
<pre class="r"><code>library(phyloseq)
mdata_phy_all &lt;- tax_glom(ps_slv_filt_AVG, taxrank = &quot;Class&quot;, NArm = FALSE)
# You can choose any taxonomic level here
mdata_phyrel_all &lt;- transform_sample_counts(
  mdata_phy_all, function(x) x / sum(x)
  )
meltd_all &lt;- psmelt(mdata_phyrel_all)
meltd_all$Class &lt;- as.character(meltd_all$Class)

means &lt;- ddply(meltd_all, ~Class, function(x) c(mean = mean(x$Abundance)))
# decending order
taxa_means &lt;- means[order(-means$mean), ]  
# ditch the sci notation 
taxa_means &lt;- format(taxa_means, scientific = FALSE)  

# Here we conglomerate at 2%.
Other &lt;- means[means$mean &lt;= 0.026, ]$Class  

meltd_all[meltd_all$Class %in% Other, ]$Class &lt;- &quot;Other&quot;
samp_names &lt;- aggregate(meltd_all$Abundance, 
                        by = list(meltd_all$Sample), FUN = sum)[, 1]
.e &lt;- environment()
meltd_all[, &quot;Class&quot;] &lt;- factor(meltd_all[, &quot;Class&quot;], 
                               sort(unique(meltd_all[, &quot;Class&quot;])))
meltd_all &lt;- meltd_all[order(meltd_all[, &quot;Class&quot;]), ]
levels(meltd_all$Class)

# Here we order Classes by the Phylum they belong to.
meltd_all$Class &lt;- factor(meltd_all$Class, 
                          levels = c(
                            &quot;Bacteroidia&quot;, &quot;Clostridia&quot;, &quot;Erysipelotrichia&quot;, 
                            &quot;Fusobacteriia&quot;, &quot;Alphaproteobacteria&quot;, 
                            &quot;Deltaproteobacteria&quot;, &quot;Gammaproteobacteria&quot;, 
                            &quot;Planctomycetacia&quot;, &quot;Oxyphotobacteria&quot;, &quot;Other&quot;))  

sup_fig1 &lt;- qplot(data = meltd_all, x = Sp, y = Abundance, fill = Class, 
                  geom = &quot;boxplot&quot;, ylab = &quot;Relative Abundance&quot;) +
  theme(legend.position = &quot;bottom&quot;) + 
  facet_grid(Class ~ ., scales = &quot;free_y&quot;, space = &quot;free_y&quot;) + 
  geom_jitter(width = 0.05) + 
  geom_point(colour = &quot;black&quot;, fill = &quot;white&quot;)  
#+ guides(guide_legend(reverse = FALSE) )

sup_fig1 &lt;- sup_fig1 + 
  scale_fill_manual(values = friend_pal) + 
  labs(x = &quot;Host species&quot;, y = &quot;Relative abundance (% total reads)&quot;)

pdf(&quot;DATA/PHYLOSEQ/FIGURES/OUTPUT/box_and_whisker.pdf&quot;)
sup_fig1
invisible(dev.off())</code></pre>
</div>
<div id="code-for-bar-plot" class="section level4">
<h4>Code for bar plot</h4>
<div class="paper">
Figure S1
</div>
<pre class="r"><code>sup_fig2 &lt;- ggplot(meltd_all, aes_string(x = &quot;Sample&quot;, y = &quot;Abundance&quot;, 
    fill = &quot;Class&quot;), environment = .e, Ordered = TRUE)
sup_fig2 &lt;- sup_fig2 + geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) + 
    facet_grid(Class ~ Sp, scales = &quot;free&quot;, space = &quot;free&quot;)
sup_fig2 &lt;- sup_fig2 + scale_fill_manual(values = friend_pal)

# sup_fig2 &lt;- sup_fig2 + theme(axis.text.x = element_text(angle = -90,
# hjust = 0))
sup_fig2 &lt;- sup_fig2 + theme(axis.text.x = element_blank())

sup_fig2 &lt;- sup_fig2 + 
  guides(fill = guide_legend(override.aes = list(colour = NULL), 
                             reverse = FALSE)) + 
  theme(legend.key = element_rect(colour = &quot;black&quot;), 
        legend.position = &quot;bottom&quot;) + 
  labs(x = &quot;Individual samples&quot;, y = &quot;Relative abundance (% total reads)&quot;)

pdf(&quot;DATA/PHYLOSEQ/FIGURES/OUTPUT/Figure_S1.pdf&quot;)
sup_fig2
invisible(dev.off())</code></pre>
<pre class="r"><code>grid.arrange(sup_fig1, sup_fig2, ncol = 2)</code></pre>
<div class="figure" style="text-align: center">
<img src="7_appendices_files/figure-html/grid.arrange_plots-1.png" alt="Figure S1. Relative abundance of major Classes by sample" width="90%" />
<p class="caption">
Figure S1. Relative abundance of major Classes by sample
</p>
</div>
</div>
</div>
<div id="appendix-b-tools-resources-used-in-this-workflow" class="section level2">
<h2>Appendix B: Tools &amp; resources used in this workflow</h2>
<div id="specific-tools" class="section level4">
<h4>Specific tools</h4>
<p>– <a href="https://joey711.github.io/phyloseq/" target="_blank">phyloseq</a> as the primary analytical package.<br />
– <a href="https://dx.doi.org/10.1038%2Fnmeth.2658" target="_blank">LEfSe</a> to identify differentially abundant (DA) amplicon sequence variants (ASV) across host fish species.<br />
– <a href="http://www.microbiomeanalyst.ca/" target="_blank">MicrobiomeAnalyst</a> to conduct LEfSe analysis.<br />
– <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank">BLASTn</a> and <a href="https://www.arb-silva.de/aligner/" target="_blank">Silva ACT</a> to identify closest hits to DA ASVs.<br />
– <a href="https://sco.h-its.org/exelixis/web/software/raxml/" target="_blank">RAxML</a> for phylogenetic inference of DA ASVs and closest hits.<br />
– <a href="https://itol.embl.de/" target="_blank">iTOL</a> for visualization of tree and associated metadata.</p>
</div>
<div id="other-valuable-resources" class="section level4">
<h4>Other valuable resources</h4>
<p>– <a href="https://bookdown.org/yihui/rmarkdown/" target="_blank">R Markdown: The Definitive Guide</a><br />
– <a href="http://yihui.name/knitr/" target="_blank">knitr</a> tutorials. Fantastic site!<br />
– <a href="https://rpubs.com/maddieSC/R_SOP_UCR_Jan_2018" target="_blank">Microbiota analysis in R, UCR Workshop 2018</a> nicely documented workflow with examples.</p>
</div>
</div>
<div id="appendix-c-submitting-sequencing-data-to-public-archives" class="section level2">
<h2>Appendix C: Submitting sequencing data to public archives</h2>
<p>It is now time to submit the data to your favorite sequence read archive. We submitted out data to the <a href="https://www.ebi.ac.uk/ena" target="blank">European Nucleotide Archive (ENA)</a>. The ENA does not like RAW data and prefers to have primers removed. So we submitted the trimmed Fastq files to the ENA. You can find these data under the study accession number <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB28397" target="blank"><strong>PRJEB28397</strong></a>. The raw, raw files are available on the project’s <a href="https://doi.org/10.6084/m9.figshare.6875522" target="blank">figshare site</a>.</p>
<p>To submit to the ENA you need two data tables (plus your sequence data). The first file describes the samples and the second file describes the sequencing data. The original files can be found on the figshare site with the <a href="https://doi.org/10.6084/m9.figshare.6875522" target="_blank">raw data</a>.</p>
</div>
<div id="appendix-d-specific-r-package-versions" class="section level2">
<h2>Appendix D: Specific R package &amp; versions</h2>
<p>Below are the specific packages and versions used in this workflow using both <code>sessionInfo()</code> and <code>devtools::session_info()</code>.</p>
<pre class="r"><code>proc.time() - ptm</code></pre>
<pre><code>##      user    system   elapsed 
## -4967.706   -39.278 -1093.409</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## Random number generation:
##  RNG:     Mersenne-Twister 
##  Normal:  Inversion 
##  Sample:  Rounding 
##  
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] phyloseq_1.28.0             treeio_1.8.2               
##  [3] tidytree_0.2.7              ggtree_1.16.6              
##  [5] picante_1.8                 nlme_3.1-141               
##  [7] gdata_2.18.0                heatmap3_1.1.6             
##  [9] circlize_0.4.8              ComplexHeatmap_2.1.0       
## [11] ggpubr_0.2.3                magrittr_1.5               
## [13] ggthemes_4.2.0              seqinr_3.6-1               
## [15] ips_0.0.11                  ape_5.3                    
## [17] TeachingDemos_2.10          pairwiseAdonis_0.0.1       
## [19] cluster_2.1.0               plotly_4.9.0               
## [21] RCurl_1.95-4.12             bitops_1.0-6               
## [23] svgPanZoom_0.3.3            gridExtra_2.3              
## [25] forcats_0.4.0               stringr_1.4.0              
## [27] dplyr_0.8.3                 purrr_0.3.2                
## [29] readr_1.3.1                 tidyr_0.8.3                
## [31] tibble_2.1.3                tidyverse_1.2.1            
## [33] formatR_1.7                 pander_0.6.3               
## [35] rmarkdown_1.16              DT_0.8                     
## [37] data.table_1.12.2           kableExtra_1.1.0           
## [39] knitr_1.25                  rstudioapi_0.10            
## [41] reshape2_1.4.3              scales_1.0.0               
## [43] vegan_2.5-6                 lattice_0.20-38            
## [45] permute_0.9-5               plyr_1.8.4                 
## [47] ggplot2_3.2.1               ShortRead_1.42.0           
## [49] GenomicAlignments_1.20.1    SummarizedExperiment_1.14.1
## [51] DelayedArray_0.10.0         matrixStats_0.55.0         
## [53] Biobase_2.44.0              Rsamtools_2.0.0            
## [55] GenomicRanges_1.36.1        GenomeInfoDb_1.20.0        
## [57] Biostrings_2.52.0           XVector_0.24.0             
## [59] IRanges_2.18.2              S4Vectors_0.22.1           
## [61] BiocParallel_1.18.1         BiocGenerics_0.30.0        
## [63] dada2_1.12.1                Rcpp_1.0.2                 
## 
## loaded via a namespace (and not attached):
##  [1] readxl_1.3.1           backports_1.1.5        fastmatch_1.1-0       
##  [4] igraph_1.2.4.1         lazyeval_0.2.2         splines_3.6.0         
##  [7] crosstalk_1.0.0        usethis_1.5.1          digest_0.6.21         
## [10] foreach_1.4.7          htmltools_0.4.0        memoise_1.1.0         
## [13] remotes_2.1.0          fastcluster_1.1.25     modelr_0.1.5          
## [16] RcppParallel_4.4.3     prettyunits_1.0.2      colorspace_1.4-1      
## [19] rvest_0.3.4            haven_2.1.1            xfun_0.10             
## [22] callr_3.3.1            crayon_1.3.4           jsonlite_1.6          
## [25] zeallot_0.1.0          survival_2.44-1.1      phangorn_2.5.5        
## [28] iterators_1.0.12       glue_1.3.1             gtable_0.3.0          
## [31] zlibbioc_1.30.0        webshot_0.5.1          GetoptLong_0.1.7      
## [34] pkgbuild_1.0.5         Rhdf5lib_1.6.1         shape_1.4.4           
## [37] xtable_1.8-4           viridisLite_0.3.0      clue_0.3-57           
## [40] htmlwidgets_1.3        httr_1.4.1             RColorBrewer_1.1-2    
## [43] ellipsis_0.3.0         pkgconfig_2.0.3        XML_3.98-1.20         
## [46] labeling_0.3           later_1.0.0            tidyselect_0.2.5      
## [49] rlang_0.4.0            munsell_0.5.0          cellranger_1.1.0      
## [52] tools_3.6.0            cli_1.1.0              generics_0.0.2        
## [55] ade4_1.7-13            devtools_2.2.0         broom_0.5.2           
## [58] evaluate_0.14          biomformat_1.12.0      yaml_2.2.0            
## [61] processx_3.4.1         fs_1.3.1               mime_0.7              
## [64] xml2_1.2.2             compiler_3.6.0         png_0.1-7             
## [67] ggsignif_0.6.0         testthat_2.2.1         stringi_1.4.3         
## [70] highr_0.8              ps_1.3.0               desc_1.2.0            
## [73] Matrix_1.2-17          multtest_2.40.0        vctrs_0.2.0           
## [76] pillar_1.4.2           GlobalOptions_0.1.0    httpuv_1.5.2          
## [79] R6_2.4.0               latticeExtra_0.6-28    hwriter_1.3.2         
## [82] promises_1.1.0         sessioninfo_1.1.1      codetools_0.2-16      
## [85] gtools_3.8.1           MASS_7.3-51.4          assertthat_0.2.1      
## [88] pkgload_1.0.2          rhdf5_2.28.0           rjson_0.2.20          
## [91] rprojroot_1.3-2        withr_2.1.2            GenomeInfoDbData_1.2.1
## [94] mgcv_1.8-28            hms_0.5.1              quadprog_1.5-7        
## [97] rvcheck_0.1.3          shiny_1.3.2            lubridate_1.7.4</code></pre>
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 3.6.0 (2019-04-26)
##  os       macOS Mojave 10.14.6        
##  system   x86_64, darwin15.6.0        
##  ui       RStudio                     
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       America/Panama              
##  date     2019-10-17                  
## 
## ─ Packages ──────────────────────────────────────────────────────────────
##  package              * version   date       lib
##  ade4                   1.7-13    2018-08-31 [1]
##  ape                  * 5.3       2019-03-17 [1]
##  assertthat             0.2.1     2019-03-21 [1]
##  backports              1.1.5     2019-10-02 [1]
##  Biobase              * 2.44.0    2019-05-02 [1]
##  BiocGenerics         * 0.30.0    2019-05-02 [1]
##  BiocParallel         * 1.18.1    2019-08-06 [1]
##  biomformat             1.12.0    2019-05-02 [1]
##  Biostrings           * 2.52.0    2019-05-02 [1]
##  bitops               * 1.0-6     2013-08-17 [1]
##  broom                  0.5.2     2019-04-07 [1]
##  callr                  3.3.1     2019-07-18 [1]
##  cellranger             1.1.0     2016-07-27 [1]
##  circlize             * 0.4.8     2019-09-08 [1]
##  cli                    1.1.0     2019-03-19 [1]
##  clue                   0.3-57    2019-02-25 [1]
##  cluster              * 2.1.0     2019-06-19 [1]
##  codetools              0.2-16    2018-12-24 [1]
##  colorspace             1.4-1     2019-03-18 [1]
##  ComplexHeatmap       * 2.1.0     2019-09-10 [1]
##  crayon                 1.3.4     2017-09-16 [1]
##  crosstalk              1.0.0     2016-12-21 [1]
##  dada2                * 1.12.1    2019-05-14 [1]
##  data.table           * 1.12.2    2019-04-07 [1]
##  DelayedArray         * 0.10.0    2019-05-02 [1]
##  desc                   1.2.0     2018-05-01 [1]
##  devtools               2.2.0     2019-09-07 [1]
##  digest                 0.6.21    2019-09-20 [1]
##  dplyr                * 0.8.3     2019-07-04 [1]
##  DT                   * 0.8       2019-08-07 [1]
##  ellipsis               0.3.0     2019-09-20 [1]
##  evaluate               0.14      2019-05-28 [1]
##  fastcluster            1.1.25    2018-06-07 [1]
##  fastmatch              1.1-0     2017-01-28 [1]
##  forcats              * 0.4.0     2019-02-17 [1]
##  foreach                1.4.7     2019-07-27 [1]
##  formatR              * 1.7       2019-06-11 [1]
##  fs                     1.3.1     2019-05-06 [1]
##  gdata                * 2.18.0    2017-06-06 [1]
##  generics               0.0.2     2018-11-29 [1]
##  GenomeInfoDb         * 1.20.0    2019-05-02 [1]
##  GenomeInfoDbData       1.2.1     2019-09-10 [1]
##  GenomicAlignments    * 1.20.1    2019-06-18 [1]
##  GenomicRanges        * 1.36.1    2019-09-06 [1]
##  GetoptLong             0.1.7     2018-06-10 [1]
##  ggplot2              * 3.2.1     2019-08-10 [1]
##  ggpubr               * 0.2.3     2019-09-03 [1]
##  ggsignif               0.6.0     2019-08-08 [1]
##  ggthemes             * 4.2.0     2019-05-13 [1]
##  ggtree               * 1.16.6    2019-08-26 [1]
##  GlobalOptions          0.1.0     2018-06-09 [1]
##  glue                   1.3.1     2019-03-12 [1]
##  gridExtra            * 2.3       2017-09-09 [1]
##  gtable                 0.3.0     2019-03-25 [1]
##  gtools                 3.8.1     2018-06-26 [1]
##  haven                  2.1.1     2019-07-04 [1]
##  heatmap3             * 1.1.6     2019-03-22 [1]
##  highr                  0.8       2019-03-20 [1]
##  hms                    0.5.1     2019-08-23 [1]
##  htmltools              0.4.0     2019-10-04 [1]
##  htmlwidgets            1.3       2018-09-30 [1]
##  httpuv                 1.5.2     2019-09-11 [1]
##  httr                   1.4.1     2019-08-05 [1]
##  hwriter                1.3.2     2014-09-10 [1]
##  igraph                 1.2.4.1   2019-04-22 [1]
##  ips                  * 0.0.11    2019-07-04 [1]
##  IRanges              * 2.18.2    2019-08-24 [1]
##  iterators              1.0.12    2019-07-26 [1]
##  jsonlite               1.6       2018-12-07 [1]
##  kableExtra           * 1.1.0     2019-03-16 [1]
##  knitr                * 1.25      2019-09-18 [1]
##  labeling               0.3       2014-08-23 [1]
##  later                  1.0.0     2019-10-04 [1]
##  lattice              * 0.20-38   2018-11-04 [1]
##  latticeExtra           0.6-28    2016-02-09 [1]
##  lazyeval               0.2.2     2019-03-15 [1]
##  lubridate              1.7.4     2018-04-11 [1]
##  magrittr             * 1.5       2014-11-22 [1]
##  MASS                   7.3-51.4  2019-03-31 [1]
##  Matrix                 1.2-17    2019-03-22 [1]
##  matrixStats          * 0.55.0    2019-09-07 [1]
##  memoise                1.1.0     2017-04-21 [1]
##  mgcv                   1.8-28    2019-03-21 [1]
##  mime                   0.7       2019-06-11 [1]
##  modelr                 0.1.5     2019-08-08 [1]
##  multtest               2.40.0    2019-05-02 [1]
##  munsell                0.5.0     2018-06-12 [1]
##  nlme                 * 3.1-141   2019-08-01 [1]
##  pairwiseAdonis       * 0.0.1     2019-09-10 [1]
##  pander               * 0.6.3     2018-11-06 [1]
##  permute              * 0.9-5     2019-03-12 [1]
##  phangorn               2.5.5     2019-06-19 [1]
##  phyloseq             * 1.28.0    2019-05-02 [1]
##  picante              * 1.8       2019-03-21 [1]
##  pillar                 1.4.2     2019-06-29 [1]
##  pkgbuild               1.0.5     2019-08-26 [1]
##  pkgconfig              2.0.3     2019-09-22 [1]
##  pkgload                1.0.2     2018-10-29 [1]
##  plotly               * 4.9.0     2019-04-10 [1]
##  plyr                 * 1.8.4     2016-06-08 [1]
##  png                    0.1-7     2013-12-03 [1]
##  prettyunits            1.0.2     2015-07-13 [1]
##  processx               3.4.1     2019-07-18 [1]
##  promises               1.1.0     2019-10-04 [1]
##  ps                     1.3.0     2018-12-21 [1]
##  purrr                * 0.3.2     2019-03-15 [1]
##  quadprog               1.5-7     2019-05-06 [1]
##  R6                     2.4.0     2019-02-14 [1]
##  RColorBrewer           1.1-2     2014-12-07 [1]
##  Rcpp                 * 1.0.2     2019-07-25 [1]
##  RcppParallel           4.4.3     2019-05-22 [1]
##  RCurl                * 1.95-4.12 2019-03-04 [1]
##  readr                * 1.3.1     2018-12-21 [1]
##  readxl                 1.3.1     2019-03-13 [1]
##  remotes                2.1.0     2019-06-24 [1]
##  reshape2             * 1.4.3     2017-12-11 [1]
##  rhdf5                  2.28.0    2019-05-02 [1]
##  Rhdf5lib               1.6.1     2019-09-09 [1]
##  rjson                  0.2.20    2018-06-08 [1]
##  rlang                  0.4.0     2019-06-25 [1]
##  rmarkdown            * 1.16      2019-10-01 [1]
##  rprojroot              1.3-2     2018-01-03 [1]
##  Rsamtools            * 2.0.0     2019-05-02 [1]
##  rstudioapi           * 0.10      2019-03-19 [1]
##  rvcheck                0.1.3     2018-12-06 [1]
##  rvest                  0.3.4     2019-05-15 [1]
##  S4Vectors            * 0.22.1    2019-09-09 [1]
##  scales               * 1.0.0     2018-08-09 [1]
##  seqinr               * 3.6-1     2019-09-07 [1]
##  sessioninfo            1.1.1     2018-11-05 [1]
##  shape                  1.4.4     2018-02-07 [1]
##  shiny                  1.3.2     2019-04-22 [1]
##  ShortRead            * 1.42.0    2019-05-02 [1]
##  stringi                1.4.3     2019-03-12 [1]
##  stringr              * 1.4.0     2019-02-10 [1]
##  SummarizedExperiment * 1.14.1    2019-07-31 [1]
##  survival               2.44-1.1  2019-04-01 [1]
##  svgPanZoom           * 0.3.3     2016-09-26 [1]
##  TeachingDemos        * 2.10      2016-02-12 [1]
##  testthat               2.2.1     2019-07-25 [1]
##  tibble               * 2.1.3     2019-06-06 [1]
##  tidyr                * 0.8.3     2019-03-01 [1]
##  tidyselect             0.2.5     2018-10-11 [1]
##  tidytree             * 0.2.7     2019-09-12 [1]
##  tidyverse            * 1.2.1     2017-11-14 [1]
##  treeio               * 1.8.2     2019-08-21 [1]
##  usethis                1.5.1     2019-07-04 [1]
##  vctrs                  0.2.0     2019-07-05 [1]
##  vegan                * 2.5-6     2019-09-01 [1]
##  viridisLite            0.3.0     2018-02-01 [1]
##  webshot                0.5.1     2018-09-28 [1]
##  withr                  2.1.2     2018-03-15 [1]
##  xfun                   0.10      2019-10-01 [1]
##  XML                    3.98-1.20 2019-06-06 [1]
##  xml2                   1.2.2     2019-08-09 [1]
##  xtable                 1.8-4     2019-04-21 [1]
##  XVector              * 0.24.0    2019-05-02 [1]
##  yaml                   2.2.0     2018-07-25 [1]
##  zeallot                0.1.0     2018-01-28 [1]
##  zlibbioc               1.30.0    2019-05-02 [1]
##  source                                         
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  Bioconductor                                   
##  Bioconductor                                   
##  Bioconductor                                   
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Github (jokergoo/ComplexHeatmap@35d1d20)       
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  Bioconductor                                   
##  Bioconductor                                   
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Github (pmartinezarbizu/pairwiseAdonis@6e09713)
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
##  CRAN (R 3.6.0)                                 
##  CRAN (R 3.6.0)                                 
##  Bioconductor                                   
## 
## [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
<pre class="r"><code>end_time &lt;- Sys.time()
end_time - start_time</code></pre>
<pre><code>## Time difference of 2.127318 hours</code></pre>
</div>

<footer role="contentinfo" id="site-footer">

<hr>

<!-- /.bottom-menu -->
	<p class="copyright">&#169; 2019 metacrobe & ProjectDIGEST,
		powered by <a href="http://www.r-project.org">R</a> + <a href="http://rmarkdown.rstudio.com">RMarkdown</a> + <a href="http://github.com">Github</a>.</p>
</footer>

<!--this prevents a white space at bottom of page -->
<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;"></div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
